// tdesign-vue-next/esm/tree/hooks/useItemState.js
/**
 * tdesign v1.19.0
 * (c) 2026 tdesign
 * @license MIT
 */

import { toRefs, ref } from 'vue';
import 'lodash-es';
import '@babel/runtime/helpers/toConsumableArray';
import '@babel/runtime/helpers/typeof';
import '../../_chunks/dep-c6946a32.js';
import '../../config-provider/hooks/useConfig.js';
import '@babel/runtime/helpers/slicedToArray';
import '../../_chunks/dep-99c7c106.js';
import '@babel/runtime/helpers/defineProperty';
import 'tdesign-icons-vue-next';
import '../../checkbox/index.js';
import '../../loading/index.js';
import '@babel/runtime/helpers/asyncToGenerator';
import '@babel/runtime/helpers/classCallCheck';
import '@babel/runtime/helpers/createClass';
import '@babel/runtime/regenerator';
import '../../_chunks/dep-dfdddbee.js';
import '../../config-provider/utils/context.js';
import '../../_chunks/dep-05b0c60d.js';
import 'dayjs';
import '../../checkbox/checkbox.js';
import '../../checkbox/props.js';
import '../../_chunks/dep-3c0a50fc.js';
import '../../_chunks/dep-8e37f9ff.js';
import '../../_chunks/dep-7c147906.js';
import '../../_chunks/dep-c219912f.js';
import '../../_chunks/dep-5483f5bb.js';
import '../../_chunks/dep-f46a0321.js';
import '../../_chunks/dep-8e8536f7.js';
import '../../_chunks/dep-13dae06b.js';
import '../../_chunks/dep-e8c7d452.js';
import '../../_chunks/dep-9b3b27fd.js';
import '../../checkbox/constants/index.js';
import '../../checkbox/hooks/useCheckboxLazyLoad.js';
import '../../_chunks/dep-63827a31.js';
import '../../checkbox/hooks/useKeyboardEvent.js';
import '../../_chunks/dep-eff1464e.js';
import '../../checkbox/group.js';
import '../../checkbox/checkbox-group-props.js';
import '../../_chunks/dep-cad87e82.js';
import '../../_chunks/dep-deff93f4.js';
import 'tdesign-vue-next/esm/common/style/web/components/checkbox/_index.less';
import '../../loading/directive.js';
import '../../loading/plugin.js';
import '../../_chunks/dep-09f95d07.js';
import '../../loading/icon/gradient.js';
import '../../_chunks/dep-00cd43cd.js';
import '@babel/runtime/helpers/objectWithoutProperties';
import '../../loading/props.js';
import 'tdesign-vue-next/esm/common/style/web/components/loading/_index.less';

function useItemState(props, context) {
  var refProps = toRefs(props);
  var treeItemRef = ref(null);
  var treeScope = props.treeScope;
  var store = treeScope.store;
  var node = store.privateMap.get(props.itemKey);
  var state = {
    stateId: props.stateId,
    props: props,
    context: context,
    treeScope: treeScope,
    refProps: refProps,
    node: node,
    treeItemRef: treeItemRef
  };
  return {
    state: state
  };
}

export { useItemState as default };
