// tdesign-vue-next/cjs/menu/head-menu.js
/**
 * tdesign v1.19.0
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var Vue = require('vue');
var _toConsumableArray = require('@babel/runtime/helpers/toConsumableArray');
var _slicedToArray = require('@babel/runtime/helpers/slicedToArray');
var tdesignIconsVueNext = require('tdesign-icons-vue-next');
var log = require('../_chunks/dep-fac69cec.js');
var menu_headMenuProps = require('./head-menu-props.js');
var tabs_index = require('../tabs/index.js');
var menu_submenu = require('./submenu.js');
var menu_components_popupOverflowContent = require('./components/popup-overflow-content.js');
var menu_utils_vMenu = require('./utils/v-menu.js');
require('@babel/runtime/helpers/typeof');
require('../_chunks/dep-5de13842.js');
var index = require('../_chunks/dep-4212631b.js');
var index$2 = require('../_chunks/dep-3ff65181.js');
require('../_chunks/dep-31b1e4d7.js');
var index$3 = require('../_chunks/dep-df0a6b30.js');
var index$1 = require('../_chunks/dep-721bc27c.js');
var isArray = require('../_chunks/dep-ac99965c.js');
var isFunction = require('../_chunks/dep-f4e9436d.js');
require('../tabs/tabs.js');
require('../tabs/tab-panel.js');
require('../tabs/tab-panel-props.js');
require('../_chunks/dep-d522f199.js');
require('../_chunks/dep-ee9debba.js');
require('../_chunks/dep-1ad2ac36.js');
require('../_chunks/dep-04f0c113.js');
require('../_chunks/dep-c0f98bf1.js');
require('../_chunks/dep-919c39e9.js');
require('../_chunks/dep-e9164e20.js');
require('../_chunks/dep-1f93a4fb.js');
require('../_chunks/dep-8602102b.js');
require('../_chunks/dep-36c3ff8a.js');
require('../_chunks/dep-30b893ce.js');
require('../_chunks/dep-29a6a0fe.js');
require('../_chunks/dep-19fab953.js');
require('../_chunks/dep-5fd85b1e.js');
require('../config-provider/hooks/useConfig.js');
require('../_chunks/dep-8408bcc7.js');
require('../_chunks/dep-a9d730be.js');
require('../_chunks/dep-a3769001.js');
require('dayjs');
require('../_chunks/dep-eb5fcbdf.js');
require('../_chunks/dep-83a70289.js');
require('../_chunks/dep-5794541b.js');
require('../_chunks/dep-b163b14a.js');
require('../_chunks/dep-f3b0c370.js');
require('../_chunks/dep-de6498e2.js');
require('../_chunks/dep-20f60e28.js');
require('../_chunks/dep-054cd750.js');
require('../_chunks/dep-e708216c.js');
require('../_chunks/dep-6d7cbeac.js');
require('../_chunks/dep-49e8331f.js');
require('../_chunks/dep-455c76a3.js');
require('../_chunks/dep-ef0f6cb2.js');
require('../_chunks/dep-4dca9fa4.js');
require('../_chunks/dep-ab2c6a84.js');
require('../_chunks/dep-f807a184.js');
require('../tabs/tab-nav.js');
require('../tabs/props.js');
require('../tabs/tab-nav-item.js');
require('../_chunks/dep-bdb76501.js');
require('../_chunks/dep-3c0b81a2.js');
require('../_chunks/dep-d6320ff5.js');
require('../tabs/tab-nav-bar.js');
require('../_chunks/dep-3a21079e.js');
require('../_chunks/dep-9c51e332.js');
require('../_chunks/dep-76a7e7f9.js');
require('../_chunks/dep-6d02741a.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('../_chunks/dep-73da1c1e.js');
require('./submenu-props.js');
require('../common-components/fake-arrow.js');
require('../_chunks/dep-c0d22f5a.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('../_chunks/dep-0b181efe.js');
require('../_chunks/dep-e2ef7eaa.js');
require('../_chunks/dep-8e6a1720.js');
require('../popup/index.js');
require('../popup/popup.js');
require('@popperjs/core');
require('../popup/container.js');
require('../popup/props.js');
require('../_chunks/dep-2a9e99e2.js');
require('../_chunks/dep-ed830a31.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);
var _toConsumableArray__default = /*#__PURE__*/_interopDefaultLegacy(_toConsumableArray);
var _slicedToArray__default = /*#__PURE__*/_interopDefaultLegacy(_slicedToArray);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty__default["default"](e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _createForOfIteratorHelper(r, e) { var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (!t) { if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) { t && (r = t); var _n = 0, F = function F() {}; return { s: F, n: function n() { return _n >= r.length ? { done: !0 } : { done: !1, value: r[_n++] }; }, e: function e(r) { throw r; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var o, a = !0, u = !1; return { s: function s() { t = t.call(r); }, n: function n() { var r = t.next(); return a = r.done, r; }, e: function e(r) { u = !0, o = r; }, f: function f() { try { a || null == t["return"] || t["return"](); } finally { if (u) throw o; } } }; }
function _unsupportedIterableToArray(r, a) { if (r) { if ("string" == typeof r) return _arrayLikeToArray(r, a); var t = {}.toString.call(r).slice(8, -1); return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0; } }
function _arrayLikeToArray(r, a) { (null == a || a > r.length) && (a = r.length); for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e]; return n; }
function _isSlot(s) {
  return typeof s === 'function' || Object.prototype.toString.call(s) === '[object Object]' && !Vue.isVNode(s);
}
var MORE_SUBMENU_VALUE = "__t_head_menu_more__";
var _HeadMenu = Vue.defineComponent({
  name: "THeadMenu",
  props: menu_headMenuProps["default"],
  setup: function setup(props2, ctx) {
    var classPrefix = index.usePrefixClass();
    var _getCurrentInstance = Vue.getCurrentInstance(),
      proxy = _getCurrentInstance.proxy;
    Vue.watchEffect(function () {
      if (ctx.slots.options) {
        log.log.warnOnce("TMenu", "`options` slot is going to be deprecated, please use `operations` for slot instead.");
      }
    });
    var _toRefs = Vue.toRefs(props2),
      value = _toRefs.value,
      modelValue = _toRefs.modelValue,
      expanded = _toRefs.expanded;
    var _useVModel = index$1.useVModel(value, modelValue, props2.defaultValue, props2.onChange),
      _useVModel2 = _slicedToArray__default["default"](_useVModel, 2),
      activeValue = _useVModel2[0],
      setActiveValue = _useVModel2[1];
    var _useDefaultValue = index$2.useDefaultValue(expanded, props2.defaultExpanded, props2.onExpand, "expanded"),
      _useDefaultValue2 = _slicedToArray__default["default"](_useDefaultValue, 2),
      expandValues = _useDefaultValue2[0],
      setExpanded = _useDefaultValue2[1];
    var activeValues = Vue.ref([]);
    var theme = Vue.computed(function () {
      return props2.theme;
    });
    var menuClass = Vue.computed(function () {
      return ["".concat(classPrefix.value, "-menu"), "".concat(classPrefix.value, "-head-menu"), "".concat(classPrefix.value, "-menu--").concat(props2.theme)];
    });
    var mode = Vue.ref(props2.expandType);
    var submenu = Vue.reactive([]);
    var vMenu = new menu_utils_vMenu.VMenu({
      isMutex: true,
      expandValues: expandValues.value
    });
    Vue.provide("TdMenu", {
      mode: mode,
      theme: theme,
      vMenu: vMenu,
      isHead: true,
      expandValues: expandValues,
      activeValue: activeValue,
      activeValues: activeValues,
      select: function select(value2) {
        setActiveValue(value2);
      },
      open: function open(value2, type) {
        var expanded2 = _toConsumableArray__default["default"](expandValues.value);
        var index = expanded2.indexOf(value2);
        if (mode.value === "popup") {
          if (type === "add") {
            if (index === -1) {
              expanded2.push(value2);
            }
          } else if (type === "remove") {
            expanded2.splice(index, 1);
          }
        } else if (mode.value === "normal" && value2 !== void 0) {
          expanded2.splice(0, 1);
          if (index === -1) {
            expanded2.push(value2);
          }
        }
        setExpanded(expanded2);
      }
    });
    var handleTabChange = function handleTabChange(value2) {
      setActiveValue(value2);
      handleClickSubMenuItem(value2);
    };
    var handleSubmenuExpand = function handleSubmenuExpand(value2) {
      var ans = vMenu.getChild(value2);
      submenu.length = 0;
      submenu.push.apply(submenu, _toConsumableArray__default["default"](ans));
    };
    Vue.watch(expandValues, function (value2) {
      if (mode.value === "normal") {
        handleSubmenuExpand(value2[0]);
      }
    });
    var updateActiveValues = function updateActiveValues(value2) {
      activeValues.value = vMenu.select(value2);
      syncMoreActiveState();
    };
    var syncMoreActiveState = function syncMoreActiveState() {
      var vals = activeValues.value;
      var hasMoreFlag = vals.includes(MORE_SUBMENU_VALUE);
      var isFolded = foldStartIndex.value >= 0;
      var needActive = false;
      if (isFolded && vals.length > 0) {
        var _vMenu$data;
        var topLevelValues = ((_vMenu$data = vMenu.data) === null || _vMenu$data === void 0 || (_vMenu$data = _vMenu$data.children) === null || _vMenu$data === void 0 ? void 0 : _vMenu$data.map(function (child) {
          return child.value;
        })) || [];
        var foldedTopValues = new Set(topLevelValues.slice(foldStartIndex.value));
        needActive = vals.some(function (v) {
          return v != null && v !== MORE_SUBMENU_VALUE && foldedTopValues.has(v);
        });
      }
      if (needActive && !hasMoreFlag) {
        activeValues.value = [].concat(_toConsumableArray__default["default"](vals), [MORE_SUBMENU_VALUE]);
      } else if (!needActive && hasMoreFlag) {
        activeValues.value = vals.filter(function (v) {
          return v !== MORE_SUBMENU_VALUE;
        });
      }
    };
    Vue.watch(activeValue, updateActiveValues);
    Vue.watch(function () {
      return props2.expandType;
    }, function (value2) {
      mode.value = value2;
    });
    Vue.onMounted(function () {
      var _expandValues$value;
      activeValues.value = vMenu.select(activeValue.value);
      if (((_expandValues$value = expandValues.value) === null || _expandValues$value === void 0 ? void 0 : _expandValues$value.length) > 0) {
        handleSubmenuExpand(expandValues.value[0]);
      }
    });
    var handleClickSubMenuItem = function handleClickSubMenuItem(value2) {
      var _activeMenuItem$onCli;
      var activeMenuItem = submenu.find(function (v) {
        return v.value === value2;
      });
      (_activeMenuItem$onCli = activeMenuItem.onClick) === null || _activeMenuItem$onCli === void 0 || _activeMenuItem$onCli.call(activeMenuItem, {
        value: value2
      });
      var to = activeMenuItem.to,
        href = activeMenuItem.href,
        replace = activeMenuItem.replace;
      if (href) {
        window.location.href = activeMenuItem.href;
      }
      var router = activeMenuItem.router || proxy.$router;
      if (to && router) {
        replace ? router.replace(to) : router.push(to);
      }
    };
    var renderNormalSubmenu = function renderNormalSubmenu() {
      var _slot;
      if (submenu.length === 0) return null;
      return Vue.createVNode("ul", {
        "class": ["".concat(classPrefix.value, "-head-menu__submenu"), "".concat(classPrefix.value, "-submenu")]
      }, [Vue.createVNode(tabs_index.Tabs, {
        "value": activeValue.value,
        "onChange": handleTabChange
      }, _isSlot(_slot = submenu.map(function (item) {
        var _item$vnode$;
        return Vue.createVNode(tabs_index.TabPanel, {
          "value": item.value,
          "label": (_item$vnode$ = item.vnode()[0]) === null || _item$vnode$ === void 0 ? void 0 : _item$vnode$.children
        }, null);
      })) ? _slot : {
        "default": function _default() {
          return [_slot];
        }
      })]);
    };
    var menuRef = Vue.ref();
    var innerRef = Vue.ref();
    var logoRef = Vue.ref();
    var operationRef = Vue.ref();
    var foldStartIndex = Vue.ref(-1);
    var cachedItemWidths = [];
    var getMenuItemElements = function getMenuItemElements() {
      if (!menuRef.value) return [];
      var moreClass = "".concat(classPrefix.value, "-head-menu__submenu--more");
      var menuItemClass = "".concat(classPrefix.value, "-menu__item");
      var submenuClass = "".concat(classPrefix.value, "-submenu");
      var result = [];
      var _collectItems = function collectItems(parent, depth) {
        if (depth > 3) return;
        Array.from(parent.children).forEach(function (el) {
          if (!(el instanceof HTMLElement)) return;
          if (el.classList.contains(moreClass)) return;
          if (el.classList.contains(menuItemClass) || el.classList.contains(submenuClass)) {
            result.push(el);
          } else {
            _collectItems(el, depth + 1);
          }
        });
      };
      _collectItems(menuRef.value, 0);
      return result;
    };
    var getMoreButtonWidth = function getMoreButtonWidth() {
      if (!menuRef.value) return 0;
      var moreEl = menuRef.value.querySelector(".".concat(classPrefix.value, "-head-menu__submenu--more"));
      if (!moreEl) return 0;
      var wasHidden = moreEl.style.display === "none";
      if (wasHidden) {
        moreEl.style.visibility = "hidden";
        moreEl.style.display = "";
      }
      var style = window.getComputedStyle(moreEl);
      var rect = moreEl.getBoundingClientRect();
      var w = rect.width + Number.parseFloat(style.marginLeft) + Number.parseFloat(style.marginRight);
      if (wasHidden) {
        moreEl.style.display = "none";
        moreEl.style.visibility = "";
      }
      return w;
    };
    var getElementWidth = function getElementWidth(el) {
      var style = window.getComputedStyle(el);
      var rect = el.getBoundingClientRect();
      return rect.width + Number.parseFloat(style.marginLeft) + Number.parseFloat(style.marginRight);
    };
    var getWrapperElements = function getWrapperElements() {
      if (!menuRef.value) return [];
      var moreClass = "".concat(classPrefix.value, "-head-menu__submenu--more");
      var menuItemClass = "".concat(classPrefix.value, "-menu__item");
      var submenuClass = "".concat(classPrefix.value, "-submenu");
      var wrappers = [];
      Array.from(menuRef.value.children).forEach(function (el) {
        if (!(el instanceof HTMLElement)) return;
        if (el.classList.contains(moreClass)) return;
        if (el.classList.contains(menuItemClass) || el.classList.contains(submenuClass)) return;
        wrappers.push(el);
      });
      return wrappers;
    };
    var handleResize = function handleResize() {
      if (props2.expandType !== "popup") return;
      if (!menuRef.value) return;
      var wrappers = getWrapperElements();
      var savedWrapperDisplays = [];
      wrappers.forEach(function (el) {
        savedWrapperDisplays.push(el.style.display);
        el.style.display = "contents";
      });
      var itemNodes = getMenuItemElements();
      if (itemNodes.length === 0) {
        wrappers.forEach(function (el, i) {
          el.style.display = savedWrapperDisplays[i];
        });
        foldStartIndex.value = -1;
        applyFoldState();
        return;
      }
      var moreEl = menuRef.value.querySelector(".".concat(classPrefix.value, "-head-menu__submenu--more"));
      if (moreEl) moreEl.style.display = "none";
      var savedFlexShrinks = [];
      var savedDisplays = [];
      itemNodes.forEach(function (el) {
        savedDisplays.push(el.style.display);
        el.style.display = "";
        savedFlexShrinks.push(el.style.flexShrink);
        el.style.flexShrink = "0";
      });
      cachedItemWidths.length = 0;
      itemNodes.forEach(function (el) {
        cachedItemWidths.push(getElementWidth(el));
      });
      itemNodes.forEach(function (el, index) {
        el.style.flexShrink = savedFlexShrinks[index];
        el.style.display = savedDisplays[index];
      });
      if (moreEl) moreEl.style.display = savedDisplays.length > 0 ? "" : "none";
      wrappers.forEach(function (el) {
        el.style.display = "contents";
      });
      var menuWidth = calcMenuWidth();
      var totalItemsWidth = cachedItemWidths.reduce(function (sum, w) {
        return sum + w;
      }, 0);
      var newFoldIndex = -1;
      if (totalItemsWidth > menuWidth) {
        var subMoreWidth = getMoreButtonWidth();
        var currentWidth = 0;
        for (var i = 0; i < itemNodes.length; i++) {
          if (currentWidth + cachedItemWidths[i] + subMoreWidth > menuWidth) {
            newFoldIndex = i;
            break;
          }
          currentWidth += cachedItemWidths[i];
        }
        if (newFoldIndex === -1 && currentWidth + subMoreWidth > menuWidth) {
          newFoldIndex = itemNodes.length - 1;
        }
      }
      foldStartIndex.value = newFoldIndex;
      applyFoldState();
    };
    var applyFoldState = function applyFoldState() {
      if (!menuRef.value) return;
      var wrappers = getWrapperElements();
      wrappers.forEach(function (el) {
        el.style.display = "contents";
      });
      var itemNodes = getMenuItemElements();
      var moreEl = menuRef.value.querySelector(".".concat(classPrefix.value, "-head-menu__submenu--more"));
      var isFolded = foldStartIndex.value >= 0;
      itemNodes.forEach(function (el, index) {
        el.style.display = isFolded && index >= foldStartIndex.value ? "none" : "";
      });
      if (moreEl) {
        moreEl.style.display = isFolded ? "" : "none";
      }
      syncMoreActiveState();
    };
    index$3.useResizeObserver(innerRef, handleResize);
    Vue.watch(logoRef, function (el) {
      if (el) {
        var imgs = el.querySelectorAll("img");
        imgs.forEach(function (img) {
          if (!img.complete) {
            img.onload = function () {
              return handleResize();
            };
          }
        });
      }
    });
    var getComputedCss = function getComputedCss(el, cssProperty) {
      var _getComputedStyle$css;
      return (_getComputedStyle$css = getComputedStyle(el)[cssProperty]) !== null && _getComputedStyle$css !== void 0 ? _getComputedStyle$css : "";
    };
    var getComputedCssValue = function getComputedCssValue(el, cssProperty) {
      return Number.parseInt(String(getComputedCss(el, cssProperty)), 10);
    };
    var calcMenuWidth = function calcMenuWidth() {
      if (!innerRef.value || !menuRef.value) return 0;
      var menuPaddingLeft = getComputedCssValue(menuRef.value, "paddingLeft");
      var menuPaddingRight = getComputedCssValue(menuRef.value, "paddingRight");
      var menuMarginLeft = getComputedCssValue(menuRef.value, "marginLeft");
      var menuMarginRight = getComputedCssValue(menuRef.value, "marginRight");
      var totalWidth = innerRef.value.clientWidth;
      if (logoRef.value) {
        var logoMarginLeft = getComputedCssValue(logoRef.value, "marginLeft");
        var logoMarginRight = getComputedCssValue(logoRef.value, "marginRight");
        totalWidth = totalWidth - logoRef.value.offsetWidth - logoMarginLeft - logoMarginRight;
      }
      if (operationRef.value) {
        var operationMarginLeft = getComputedCssValue(operationRef.value, "marginLeft");
        var operationMarginRight = getComputedCssValue(operationRef.value, "marginRight");
        totalWidth = totalWidth - operationRef.value.offsetWidth - operationMarginLeft - operationMarginRight;
      }
      return totalWidth - menuPaddingLeft - menuPaddingRight - menuMarginLeft - menuMarginRight;
    };
    var _flattenSlotNodes = function flattenSlotNodes(nodes) {
      var result = [];
      var _iterator = _createForOfIteratorHelper(nodes),
        _step;
      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var _node$type, _node$type2, _node$children;
          var node = _step.value;
          if (node.type === Symbol["for"]("v-fgm") || node.type === Symbol["for"]("")) {
            if (isArray.isArray(node.children)) {
              result.push.apply(result, _toConsumableArray__default["default"](_flattenSlotNodes(node.children)));
            }
          } else if (((_node$type = node.type) === null || _node$type === void 0 ? void 0 : _node$type.name) === "TSubmenu" || ((_node$type2 = node.type) === null || _node$type2 === void 0 ? void 0 : _node$type2.name) === "TMenuItem") {
            result.push(node);
          } else if (isArray.isArray(node.children)) {
            result.push.apply(result, _toConsumableArray__default["default"](_flattenSlotNodes(node.children)));
          } else if (isFunction.isFunction((_node$children = node.children) === null || _node$children === void 0 ? void 0 : _node$children["default"])) {
            result.push.apply(result, _toConsumableArray__default["default"](_flattenSlotNodes(node.children["default"]())));
          } else {
            result.push(node);
          }
        }
      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }
      return result;
    };
    var _initVMenu = function initVMenu(slots, parentValue) {
      slots.forEach(function (node) {
        var _node$props, _node$type3, _node$type4, _node$children2;
        var nodeValue = (_node$props = node.props) === null || _node$props === void 0 ? void 0 : _node$props.value;
        if (((_node$type3 = node.type) === null || _node$type3 === void 0 ? void 0 : _node$type3.name) === "TSubmenu" || ((_node$type4 = node.type) === null || _node$type4 === void 0 ? void 0 : _node$type4.name) === "TMenuItem") {
          vMenu.add(_objectSpread({
            value: nodeValue,
            parent: parentValue,
            vnode: node.children["default"]
          }, node.props));
        }
        if (isFunction.isFunction((_node$children2 = node.children) === null || _node$children2 === void 0 ? void 0 : _node$children2["default"])) {
          _initVMenu(node.children["default"](), nodeValue);
          return;
        }
        if (isArray.isArray(node.children)) {
          _initVMenu(node.children, nodeValue);
        }
      });
    };
    var mutationObserver = null;
    var isResizing = false;
    var resizeRAF = null;
    var safeHandleResize = function safeHandleResize() {
      if (isResizing) return;
      applyFoldState();
      if (resizeRAF) cancelAnimationFrame(resizeRAF);
      resizeRAF = requestAnimationFrame(function () {
        isResizing = true;
        handleResize();
        requestAnimationFrame(function () {
          isResizing = false;
        });
      });
    };
    Vue.watch(menuRef, function (el, _oldEl) {
      if (mutationObserver) {
        mutationObserver.disconnect();
        mutationObserver = null;
      }
      if (el) {
        mutationObserver = new MutationObserver(function () {
          safeHandleResize();
        });
        mutationObserver.observe(el, {
          childList: true,
          subtree: true
        });
        Vue.nextTick(function () {
          return handleResize();
        });
      }
    });
    Vue.onBeforeUnmount(function () {
      if (mutationObserver) {
        mutationObserver.disconnect();
        mutationObserver = null;
      }
      if (resizeRAF) {
        cancelAnimationFrame(resizeRAF);
        resizeRAF = null;
      }
    });
    return function () {
      var _props2$logo, _ctx$slots$logo, _ctx$slots, _props2$operations, _ctx$slots$operations, _ctx$slots2, _ctx$slots$options, _ctx$slots3, _ctx$slots$default, _ctx$slots4, _ctx$slots$content, _ctx$slots5;
      var logo = ((_props2$logo = props2.logo) === null || _props2$logo === void 0 ? void 0 : _props2$logo.call(props2, Vue.h)) || ((_ctx$slots$logo = (_ctx$slots = ctx.slots).logo) === null || _ctx$slots$logo === void 0 ? void 0 : _ctx$slots$logo.call(_ctx$slots));
      var operations = ((_props2$operations = props2.operations) === null || _props2$operations === void 0 ? void 0 : _props2$operations.call(props2, Vue.h)) || ((_ctx$slots$operations = (_ctx$slots2 = ctx.slots).operations) === null || _ctx$slots$operations === void 0 ? void 0 : _ctx$slots$operations.call(_ctx$slots2)) || ((_ctx$slots$options = (_ctx$slots3 = ctx.slots).options) === null || _ctx$slots$options === void 0 ? void 0 : _ctx$slots$options.call(_ctx$slots3));
      var originalContent = ((_ctx$slots$default = (_ctx$slots4 = ctx.slots)["default"]) === null || _ctx$slots$default === void 0 ? void 0 : _ctx$slots$default.call(_ctx$slots4)) || ((_ctx$slots$content = (_ctx$slots5 = ctx.slots).content) === null || _ctx$slots$content === void 0 ? void 0 : _ctx$slots$content.call(_ctx$slots5)) || [];
      var flatItems = _flattenSlotNodes(originalContent);
      var isFolded = foldStartIndex.value >= 0;
      var canSliceVNodes = isFolded && foldStartIndex.value < flatItems.length;
      var content = originalContent;
      vMenu.data.children = [];
      vMenu.cache.clear();
      _initVMenu(flatItems);
      var buildPopupSlot = function buildPopupSlot() {
        var _slot2;
        if (!isFolded) return [];
        if (canSliceVNodes) {
          return flatItems.slice(foldStartIndex.value).map(function (vnode) {
            return Vue.cloneVNode(vnode);
          });
        }
        return Vue.createVNode(menu_components_popupOverflowContent["default"], {
          "foldIndex": foldStartIndex.value
        }, _isSlot(_slot2 = originalContent.map(function (vnode) {
          return Vue.cloneVNode(vnode);
        })) ? _slot2 : {
          "default": function _default() {
            return [_slot2];
          }
        });
      };
      return Vue.createVNode("div", {
        "class": menuClass.value
      }, [Vue.createVNode("div", {
        "class": "".concat(classPrefix.value, "-head-menu__inner"),
        "ref": innerRef
      }, [logo && Vue.createVNode("div", {
        "class": "".concat(classPrefix.value, "-menu__logo"),
        "ref": logoRef
      }, [logo]), Vue.createVNode("ul", {
        "class": "".concat(classPrefix.value, "-menu"),
        "ref": menuRef
      }, [content, props2.expandType === "popup" && Vue.createVNode(menu_submenu["default"], {
        "class": "".concat(classPrefix.value, "-head-menu__submenu--more"),
        "value": MORE_SUBMENU_VALUE,
        "expandType": "popup",
        "title": function title() {
          return Vue.createVNode(tdesignIconsVueNext.EllipsisIcon, null, null);
        },
        "style": {
          display: isFolded ? "" : "none"
        }
      }, {
        "default": buildPopupSlot
      })]), operations && Vue.createVNode("div", {
        "class": "".concat(classPrefix.value, "-menu__operations"),
        "ref": operationRef
      }, [operations])]), mode.value === "normal" && renderNormalSubmenu()]);
    };
  }
});

exports["default"] = _HeadMenu;
