// tdesign-vue-next/cjs/table/hooks/useColumnController.js
/**
 * tdesign v1.19.0
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var _slicedToArray = require('@babel/runtime/helpers/slicedToArray');
var _toConsumableArray = require('@babel/runtime/helpers/toConsumableArray');
var tdesignIconsVueNext = require('tdesign-icons-vue-next');
var dialog_plugin = require('../../dialog/plugin.js');
var table_hooks_useTableHeader = require('./useTableHeader.js');
require('@babel/runtime/helpers/typeof');
require('../../_chunks/dep-5de13842.js');
var index$1 = require('../../_chunks/dep-d522f199.js');
var configProvider_hooks_useConfig = require('../../config-provider/hooks/useConfig.js');
var index$2 = require('../../_chunks/dep-3ff65181.js');
var index = require('../../_chunks/dep-bdb76501.js');
require('../../_chunks/dep-31b1e4d7.js');
var table_utils_index = require('../utils/index.js');
var button_index = require('../../button/index.js');
var table_components_columnCheckboxGroup = require('../components/column-checkbox-group.js');
var isFunction = require('../../_chunks/dep-f4e9436d.js');
require('../../dialog/dialog.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('../../dialog/props.js');
require('../../_chunks/dep-4212631b.js');
require('../../_chunks/dep-f807a184.js');
require('../../_chunks/dep-182ccba7.js');
require('../../loading/icon/gradient.js');
require('../../_chunks/dep-d6320ff5.js');
require('../../_chunks/dep-c0d22f5a.js');
require('../../_chunks/dep-0b181efe.js');
require('../../_chunks/dep-e2ef7eaa.js');
require('../../_chunks/dep-1ad2ac36.js');
require('../../_chunks/dep-04f0c113.js');
require('../../_chunks/dep-ac99965c.js');
require('../../_chunks/dep-8e6a1720.js');
require('../../_chunks/dep-ee9debba.js');
require('../../_chunks/dep-5fd85b1e.js');
require('../../loading/props.js');
require('../../dialog/hooks/useSameTarget.js');
require('../../_chunks/dep-83f1b705.js');
require('../../dialog/utils/index.js');
require('../../dialog/dialog-card.js');
require('../../dialog/dialog-card-props.js');
require('../../dialog/hooks/useAction.js');
require('../../_chunks/dep-7325f3b1.js');
require('../../_chunks/dep-1f93a4fb.js');
require('../../_chunks/dep-8602102b.js');
require('../../_chunks/dep-455c76a3.js');
require('../../_chunks/dep-eb5fcbdf.js');
require('../../_chunks/dep-83a70289.js');
require('../../_chunks/dep-5794541b.js');
require('../../_chunks/dep-919c39e9.js');
require('../../_chunks/dep-b163b14a.js');
require('../../_chunks/dep-f3b0c370.js');
require('../../_chunks/dep-de6498e2.js');
require('../../_chunks/dep-20f60e28.js');
require('../../_chunks/dep-054cd750.js');
require('../../_chunks/dep-e708216c.js');
require('../../_chunks/dep-ef0f6cb2.js');
require('../../_chunks/dep-4dca9fa4.js');
require('../../_chunks/dep-ab2c6a84.js');
require('../../_chunks/dep-977d699f.js');
require('../../_chunks/dep-f55ada6c.js');
require('../../_chunks/dep-30b893ce.js');
require('../../_chunks/dep-8408bcc7.js');
require('../../_chunks/dep-a9d730be.js');
require('../../_chunks/dep-a3769001.js');
require('dayjs');
require('../../_chunks/dep-6d7cbeac.js');
require('../../_chunks/dep-49e8331f.js');
require('../../_chunks/dep-4511de03.js');
require('../../_chunks/dep-c0f98bf1.js');
require('../../_chunks/dep-e9164e20.js');
require('../../_chunks/dep-36c3ff8a.js');
require('../../_chunks/dep-29a6a0fe.js');
require('../../_chunks/dep-19fab953.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('../../button/button.js');
require('../../loading/index.js');
require('../../_chunks/dep-f201a161.js');
require('../../loading/plugin.js');
require('../../_chunks/dep-2a185a5d.js');
require('../../_chunks/dep-a262c754.js');
require('../../_chunks/dep-7fa97010.js');
require('../../_chunks/dep-e75e2fe1.js');
require('../../_chunks/dep-761e41bd.js');
require('../../_chunks/dep-cb81cbfa.js');
require('../../_chunks/dep-4e47eae6.js');
require('../../_chunks/dep-566ac4ff.js');
require('../../_chunks/dep-6d02741a.js');
require('../../button/props.js');
require('../../_chunks/dep-ec3949b4.js');
require('../../_chunks/dep-d40e34ba.js');
require('../../_chunks/dep-3c0b81a2.js');
require('./useMultiHeader.js');
require('./useClassName.js');
require('../components/ellipsis.js');
require('../../tooltip/index.js');
require('../../tooltip/tooltip.js');
require('../../tooltip/props.js');
require('../../popup/props.js');
require('../../popup/index.js');
require('../../popup/popup.js');
require('@popperjs/core');
require('../../_chunks/dep-721bc27c.js');
require('../../popup/container.js');
require('../../_chunks/dep-df0a6b30.js');
require('../../_chunks/dep-2a9e99e2.js');
require('../../_chunks/dep-9c51e332.js');
require('../../_chunks/dep-ed830a31.js');
require('../../_chunks/dep-3a21079e.js');
require('../../tooltip/utils/index.js');
require('../../checkbox/index.js');
require('../../checkbox/checkbox.js');
require('../../checkbox/props.js');
require('../../_chunks/dep-fd38990c.js');
require('../../checkbox/constants/index.js');
require('../../checkbox/hooks/useCheckboxLazyLoad.js');
require('../../_chunks/dep-c3741c2c.js');
require('../../checkbox/hooks/useKeyboardEvent.js');
require('../../_chunks/dep-78278555.js');
require('../../checkbox/group.js');
require('../../checkbox/checkbox-group-props.js');
require('../../_chunks/dep-50d2328a.js');
require('../../_chunks/dep-7f048e0f.js');
require('../../_chunks/dep-d5ee60d3.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);
var _slicedToArray__default = /*#__PURE__*/_interopDefaultLegacy(_slicedToArray);
var _toConsumableArray__default = /*#__PURE__*/_interopDefaultLegacy(_toConsumableArray);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty__default["default"](e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function getColumnKeys(columns) {
  var keys = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : /* @__PURE__ */new Set();
  for (var i = 0, len = columns.length; i < len; i++) {
    var _col$children;
    var col = columns[i];
    if ((_col$children = col.children) !== null && _col$children !== void 0 && _col$children.length) {
      getColumnKeys(col.children, keys);
    } else {
      col.colKey && keys.add(col.colKey);
    }
  }
  return keys;
}
function useColumnController(props, context) {
  var _useConfig = configProvider_hooks_useConfig.useConfig("table", props.locale),
    classPrefix = _useConfig.classPrefix,
    globalConfig = _useConfig.globalConfig;
  var _useGlobalIcon = index.useGlobalIcon({
      SettingIcon: tdesignIconsVueNext.SettingIcon
    }),
    SettingIcon = _useGlobalIcon.SettingIcon;
  var _toRefs = Vue.toRefs(props),
    columns = _toRefs.columns,
    columnController = _toRefs.columnController,
    displayColumns = _toRefs.displayColumns,
    columnControllerVisible = _toRefs.columnControllerVisible;
  var dialogInstance = Vue.ref(null);
  var renderTNodeJSX = index$1.useTNodeJSX();
  var enabledColKeys = Vue.computed(function () {
    var _columnController$val;
    var arr = (((_columnController$val = columnController.value) === null || _columnController$val === void 0 ? void 0 : _columnController$val.fields) || _toConsumableArray__default["default"](getColumnKeys(columns.value))).filter(function (v) {
      return v;
    });
    return new Set(arr);
  });
  var keys = _toConsumableArray__default["default"](getColumnKeys(columns.value));
  var _useDefaultValue = index$2.useDefaultValue(displayColumns, props.defaultDisplayColumns || keys, props.onDisplayColumnsChange, "displayColumns"),
    _useDefaultValue2 = _slicedToArray__default["default"](_useDefaultValue, 2),
    tDisplayColumns = _useDefaultValue2[0],
    setTDisplayColumns = _useDefaultValue2[1];
  var columnCheckboxKeys = Vue.ref(displayColumns.value || props.defaultDisplayColumns || keys);
  var checkboxGroupList = Vue.computed(function () {
    var _columnController$val2;
    if ((_columnController$val2 = columnController.value) !== null && _columnController$val2 !== void 0 && (_columnController$val2 = _columnController$val2.groupColumns) !== null && _columnController$val2 !== void 0 && _columnController$val2.length) {
      return getCheckboxGroupOptions(columns.value);
    }
    var oneItem = {
      label: globalConfig.value.selectAllText,
      options: getCheckboxOptions(columns.value)
    };
    return [oneItem];
  });
  Vue.watch([displayColumns], function (_ref) {
    var _ref2 = _slicedToArray__default["default"](_ref, 1),
      val = _ref2[0];
    columnCheckboxKeys.value = val || props.defaultDisplayColumns || keys;
  });
  function getOneColumnItem(column, i) {
    return {
      label: function label() {
        return table_hooks_useTableHeader.renderTitle(context.slots, column, i);
      },
      value: column.colKey,
      disabled: !enabledColKeys.value.has(column.colKey)
    };
  }
  function getCheckboxGroupOptions(columns2) {
    var _columnController$val3;
    var groupColumns = (_columnController$val3 = columnController.value) === null || _columnController$val3 === void 0 ? void 0 : _columnController$val3.groupColumns;
    if (!(groupColumns !== null && groupColumns !== void 0 && groupColumns.length)) return [];
    var groupList = [];
    var _loop = function loop(columns3) {
      for (var i = 0, len = columns3.length; i < len; i++) {
        var _column$children;
        var column = columns3[i];
        var oneItem = getOneColumnItem(column, i);
        for (var j = 0, len1 = groupColumns.length; j < len1; j++) {
          var _groupList$j;
          var groupInfo = groupColumns[j];
          if (!groupInfo.columns.includes(column.colKey)) continue;
          if ((_groupList$j = groupList[j]) !== null && _groupList$j !== void 0 && (_groupList$j = _groupList$j.options) !== null && _groupList$j !== void 0 && _groupList$j.length) {
            groupList[j].options.push(oneItem);
          } else {
            groupList[j] = _objectSpread(_objectSpread({}, groupColumns[j]), {}, {
              options: [oneItem]
            });
          }
        }
        if ((_column$children = column.children) !== null && _column$children !== void 0 && _column$children.length) {
          _loop(column.children);
        }
      }
    };
    _loop(columns2);
    return groupList;
  }
  function getCheckboxOptions(columns2) {
    var _columnController$val4;
    var arr = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : [];
    if ((_columnController$val4 = columnController.value) !== null && _columnController$val4 !== void 0 && (_columnController$val4 = _columnController$val4.groupColumns) !== null && _columnController$val4 !== void 0 && _columnController$val4.length) return [];
    for (var i = 0, len = columns2.length; i < len; i++) {
      var _item$children;
      var item = columns2[i];
      if ((_item$children = item.children) !== null && _item$children !== void 0 && _item$children.length) {
        getCheckboxOptions(item.children, arr);
      } else {
        if (item.colKey) {
          arr.push(getOneColumnItem(item, i));
        }
      }
    }
    return arr;
  }
  var handleCheckChange = function handleCheckChange(val, ctx) {
    var _props$onColumnChange;
    columnCheckboxKeys.value = val;
    var params = {
      columns: val,
      type: ctx.type,
      currentColumn: table_utils_index.getCurrentRowByKey(columns.value, String(ctx.current)),
      e: ctx.e
    };
    (_props$onColumnChange = props.onColumnChange) === null || _props$onColumnChange === void 0 || _props$onColumnChange.call(props, params);
  };
  var handleToggleColumnController = function handleToggleColumnController() {
    var _columnController$val8;
    if (columnControllerVisible.value !== void 0) {
      var _props$onColumnContro;
      (_props$onColumnContro = props.onColumnControllerVisibleChange) === null || _props$onColumnContro === void 0 || _props$onColumnContro.call(props, true, {
        trigger: "open"
      });
      context.emit("update:columnControllerVisible", true);
    }
    if (dialogInstance.value) {
      dialogInstance.value.show();
      return;
    }
    dialogInstance.value = dialog_plugin.DialogPlugin.confirm(_objectSpread({
      header: globalConfig.value.columnConfigTitleText,
      body: function body() {
        var _columnController$val5;
        var widthMode = ((_columnController$val5 = columnController.value) === null || _columnController$val5 === void 0 ? void 0 : _columnController$val5.displayType) === "fixed-width" ? "fixed" : "auto";
        var _ref3 = columnController.value || {},
          columnControllerTopContent = _ref3.columnControllerTopContent,
          columnControllerBottomContent = _ref3.columnControllerBottomContent;
        var defaultNode = Vue.createVNode("div", {
          "class": ["".concat(classPrefix.value, "-table__column-controller"), "".concat(classPrefix.value, "-table__column-controller--").concat(widthMode)]
        }, [Vue.createVNode("div", {
          "class": "".concat(classPrefix.value, "-table__column-controller-body")
        }, [isFunction.isFunction(columnControllerTopContent) ? columnControllerTopContent(Vue.h) : renderTNodeJSX("columnControllerTopContent"), globalConfig.value.columnConfigDescriptionText && Vue.createVNode("p", {
          "class": "".concat(classPrefix.value, "-table__column-controller-desc")
        }, [globalConfig.value.columnConfigDescriptionText]), checkboxGroupList.value.map(function (group, index) {
          var _columnController$val6, _columnController$val7;
          var uniqueKey = (_columnController$val6 = columnController.value) !== null && _columnController$val6 !== void 0 && (_columnController$val6 = _columnController$val6.groupColumns) !== null && _columnController$val6 !== void 0 && _columnController$val6.length ? String(group.value || index) : void 0;
          return Vue.createVNode(table_components_columnCheckboxGroup["default"], {
            "key": group.value || index,
            "uniqueKey": uniqueKey,
            "value": columnCheckboxKeys.value,
            "label": group.label,
            "options": group.options,
            "onChange": handleCheckChange,
            "checkboxProps": (_columnController$val7 = columnController.value) === null || _columnController$val7 === void 0 ? void 0 : _columnController$val7.checkboxProps
          }, null);
        }), isFunction.isFunction(columnControllerBottomContent) ? columnControllerBottomContent(Vue.h) : renderTNodeJSX("columnControllerBottomContent")])]);
        return defaultNode;
      },
      confirmBtn: globalConfig.value.confirmText,
      cancelBtn: globalConfig.value.cancelText,
      width: 612,
      onConfirm: function onConfirm() {
        setTDisplayColumns(_toConsumableArray__default["default"](columnCheckboxKeys.value));
        if (columnControllerVisible.value === void 0) {
          dialogInstance.value.hide();
        } else {
          var _props$onColumnContro2;
          (_props$onColumnContro2 = props.onColumnControllerVisibleChange) === null || _props$onColumnContro2 === void 0 || _props$onColumnContro2.call(props, false, {
            trigger: "confirm"
          });
          context.emit("update:columnControllerVisible", false);
        }
      },
      onClose: function onClose() {
        columnCheckboxKeys.value = tDisplayColumns.value;
        if (columnControllerVisible.value === void 0) {
          dialogInstance.value.hide();
        } else {
          var _props$onColumnContro3;
          (_props$onColumnContro3 = props.onColumnControllerVisibleChange) === null || _props$onColumnContro3 === void 0 || _props$onColumnContro3.call(props, false, {
            trigger: "cancel"
          });
          context.emit("update:columnControllerVisible", false);
        }
      }
    }, (_columnController$val8 = columnController.value) === null || _columnController$val8 === void 0 ? void 0 : _columnController$val8.dialogProps));
  };
  Vue.watch([columnControllerVisible], function (_ref4, _ref5) {
    var _ref6 = _slicedToArray__default["default"](_ref4, 1),
      visible = _ref6[0];
    var _ref7 = _slicedToArray__default["default"](_ref5, 1),
      oldVisible = _ref7[0];
    if (visible === void 0) return;
    if (visible === oldVisible) return;
    if (dialogInstance.value) {
      visible ? dialogInstance.value.show() : dialogInstance.value.hide();
    } else {
      visible && handleToggleColumnController();
    }
  }, {
    immediate: true
  });
  var renderColumnController = function renderColumnController() {
    var _props$columnControll;
    var isColumnController = !!(columnController.value && Object.keys(columnController.value).length);
    var placement = isColumnController ? columnController.value.placement || "top-right" : "";
    if (isColumnController && columnController.value.hideTriggerButton) return null;
    var classes = ["".concat(classPrefix.value, "-table__column-controller-trigger"), _defineProperty__default["default"]({}, "".concat(classPrefix.value, "-align-").concat(placement), !!placement)];
    return Vue.createVNode("div", {
      "class": classes
    }, [Vue.createVNode(button_index.Button, Vue.mergeProps({
      "theme": "default",
      "variant": "outline",
      "onClick": handleToggleColumnController,
      "content": globalConfig.value.columnConfigButtonText
    }, (_props$columnControll = props.columnController) === null || _props$columnControll === void 0 ? void 0 : _props$columnControll.buttonProps), {
      icon: function icon() {
        return Vue.createVNode(SettingIcon, null, null);
      }
    })]);
  };
  return {
    tDisplayColumns: tDisplayColumns,
    columnCheckboxKeys: columnCheckboxKeys,
    renderColumnController: renderColumnController
  };
}

exports["default"] = useColumnController;
exports.getColumnKeys = getColumnKeys;
