// tdesign-vue-next/cjs/drawer/drawer.js
/**
 * tdesign v1.19.0
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var tdesignIconsVueNext = require('tdesign-icons-vue-next');
require('@babel/runtime/helpers/toConsumableArray');
require('@babel/runtime/helpers/typeof');
var dom = require('../_chunks/dep-ee9debba.js');
var index$1 = require('../_chunks/dep-d522f199.js');
var index$2 = require('../_chunks/dep-4212631b.js');
var index = require('../_chunks/dep-bdb76501.js');
require('@babel/runtime/helpers/slicedToArray');
var index$3 = require('../_chunks/dep-31b1e4d7.js');
var loading_loading = require('../_chunks/dep-182ccba7.js');
var getScrollbarWidth = require('../_chunks/dep-83f1b705.js');
var drawer_props = require('./props.js');
var dialog_hooks_useAction = require('../dialog/hooks/useAction.js');
var drawer_hooks_index = require('./hooks/index.js');
var configProvider_hooks_useConfig = require('../config-provider/hooks/useConfig.js');
require('../_chunks/dep-5de13842.js');
require('../_chunks/dep-a3769001.js');
require('../_chunks/dep-f4e9436d.js');
require('../_chunks/dep-04f0c113.js');
require('../_chunks/dep-919c39e9.js');
require('../_chunks/dep-1ad2ac36.js');
require('../_chunks/dep-ac99965c.js');
require('../_chunks/dep-c0f98bf1.js');
require('../_chunks/dep-e9164e20.js');
require('../_chunks/dep-1f93a4fb.js');
require('../_chunks/dep-8602102b.js');
require('../_chunks/dep-36c3ff8a.js');
require('../_chunks/dep-30b893ce.js');
require('../_chunks/dep-29a6a0fe.js');
require('../_chunks/dep-19fab953.js');
require('../_chunks/dep-8408bcc7.js');
require('../_chunks/dep-a9d730be.js');
require('dayjs');
require('../_chunks/dep-eb5fcbdf.js');
require('../_chunks/dep-83a70289.js');
require('../_chunks/dep-5794541b.js');
require('../_chunks/dep-b163b14a.js');
require('../_chunks/dep-f3b0c370.js');
require('../_chunks/dep-de6498e2.js');
require('../_chunks/dep-20f60e28.js');
require('../_chunks/dep-054cd750.js');
require('../_chunks/dep-e708216c.js');
require('../_chunks/dep-6d7cbeac.js');
require('../_chunks/dep-49e8331f.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('../loading/icon/gradient.js');
require('../_chunks/dep-d6320ff5.js');
require('../_chunks/dep-c0d22f5a.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('../_chunks/dep-0b181efe.js');
require('../_chunks/dep-e2ef7eaa.js');
require('../_chunks/dep-8e6a1720.js');
require('../_chunks/dep-5fd85b1e.js');
require('../loading/props.js');
require('../button/index.js');
require('../button/button.js');
require('../loading/index.js');
require('../_chunks/dep-f201a161.js');
require('../loading/plugin.js');
require('../_chunks/dep-455c76a3.js');
require('../_chunks/dep-ef0f6cb2.js');
require('../_chunks/dep-4dca9fa4.js');
require('../_chunks/dep-ab2c6a84.js');
require('../_chunks/dep-2a185a5d.js');
require('../_chunks/dep-a262c754.js');
require('../_chunks/dep-7fa97010.js');
require('../_chunks/dep-e75e2fe1.js');
require('../_chunks/dep-761e41bd.js');
require('../_chunks/dep-cb81cbfa.js');
require('../_chunks/dep-977d699f.js');
require('../_chunks/dep-4e47eae6.js');
require('../_chunks/dep-566ac4ff.js');
require('../_chunks/dep-6d02741a.js');
require('../button/props.js');
require('../_chunks/dep-ec3949b4.js');
require('../_chunks/dep-d40e34ba.js');
require('../_chunks/dep-3c0b81a2.js');
require('../_chunks/dep-7325f3b1.js');
require('../_chunks/dep-f55ada6c.js');
require('../_chunks/dep-4511de03.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty__default["default"](e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var key = 1;
var _Drawer = Vue.defineComponent({
  name: "TDrawer",
  inheritAttrs: false,
  props: drawer_props["default"],
  emits: ["update:visible"],
  setup: function setup(props2, context) {
    var destroyOnCloseVisible = Vue.ref(false);
    var isVisible = Vue.ref(false);
    var styleEl = Vue.ref();
    var styleTimer = Vue.ref();
    var _useConfig = configProvider_hooks_useConfig.useConfig("drawer"),
      globalConfig = _useConfig.globalConfig;
    var _useGlobalIcon = index.useGlobalIcon({
        CloseIcon: tdesignIconsVueNext.CloseIcon
      }),
      CloseIcon = _useGlobalIcon.CloseIcon;
    var renderTNodeJSX = index$1.useTNodeJSX();
    var renderContent = index$1.useContent();
    var COMPONENT_NAME = index$2.usePrefixClass("drawer");
    var _useDrag = drawer_hooks_index.useDrag(props2),
      draggedSizeValue = _useDrag.draggedSizeValue,
      enableDrag = _useDrag.enableDrag,
      draggableLineStyles = _useDrag.draggableLineStyles,
      draggingStyles = _useDrag.draggingStyles;
    var computedVisible = Vue.computed(function () {
      return props2.visible;
    });
    var isMounted = Vue.ref(false);
    var teleportElement = loading_loading.useTeleport(function () {
      return props2.attach;
    });
    var confirmBtnAction = function confirmBtnAction(e) {
      var _props2$onConfirm;
      (_props2$onConfirm = props2.onConfirm) === null || _props2$onConfirm === void 0 || _props2$onConfirm.call(props2, {
        e: e
      });
    };
    var cancelBtnAction = function cancelBtnAction(e) {
      var _props2$onCancel;
      (_props2$onCancel = props2.onCancel) === null || _props2$onCancel === void 0 || _props2$onCancel.call(props2, {
        e: e
      });
      closeDrawer({
        trigger: "cancel",
        e: e
      });
    };
    var _useAction = dialog_hooks_useAction.useAction({
        confirmBtnAction: confirmBtnAction,
        cancelBtnAction: cancelBtnAction
      }),
      getConfirmBtn = _useAction.getConfirmBtn,
      getCancelBtn = _useAction.getCancelBtn;
    var drawerEle = Vue.ref(null);
    var drawerClasses = Vue.computed(function () {
      return [COMPONENT_NAME.value, "".concat(COMPONENT_NAME.value, "--").concat(props2.placement), _defineProperty__default["default"](_defineProperty__default["default"](_defineProperty__default["default"]({}, "".concat(COMPONENT_NAME.value, "--open"), isVisible.value), "".concat(COMPONENT_NAME.value, "--attach"), props2.showInAttachedElement), "".concat(COMPONENT_NAME.value, "--without-mask"), !props2.showOverlay), props2.drawerClassName];
    });
    var sizeValue = Vue.computed(function () {
      var _props2$size;
      if (draggedSizeValue.value) return draggedSizeValue.value;
      var size = (_props2$size = props2.size) !== null && _props2$size !== void 0 ? _props2$size : globalConfig.value.size;
      var defaultSize = isNaN(Number(size)) ? size : "".concat(size, "px");
      return {
        small: "300px",
        medium: "500px",
        large: "760px"
      }[size] || defaultSize;
    });
    var wrapperStyles = Vue.computed(function () {
      return {
        transform: isVisible.value ? "translateX(0)" : void 0,
        width: ["left", "right"].includes(props2.placement) ? sizeValue.value : "",
        height: ["top", "bottom"].includes(props2.placement) ? sizeValue.value : ""
      };
    });
    var wrapperClasses = Vue.computed(function () {
      return ["".concat(COMPONENT_NAME.value, "__content-wrapper"), "".concat(COMPONENT_NAME.value, "__content-wrapper--").concat(props2.placement)];
    });
    var parentNode = Vue.computed(function () {
      var _drawerEle$value;
      return (_drawerEle$value = drawerEle.value) === null || _drawerEle$value === void 0 ? void 0 : _drawerEle$value.parentNode;
    });
    var modeAndPlacement = Vue.computed(function () {
      return [props2.mode, props2.placement].join();
    });
    var footerStyle = Vue.computed(function () {
      return {
        display: "flex",
        justifyContent: props2.placement === "right" ? "flex-start" : "flex-end"
      };
    });
    var handleEscKeydown = function handleEscKeydown(e) {
      var _props2$closeOnEscKey;
      if (((_props2$closeOnEscKey = props2.closeOnEscKeydown) !== null && _props2$closeOnEscKey !== void 0 ? _props2$closeOnEscKey : globalConfig.value.closeOnEscKeydown) && e.key === "Escape" && isVisible.value && isTopInteractivePopup()) {
        var _props2$onEscKeydown;
        (_props2$onEscKeydown = props2.onEscKeydown) === null || _props2$onEscKeydown === void 0 || _props2$onEscKeydown.call(props2, {
          e: e
        });
        closeDrawer({
          trigger: "esc",
          e: e
        });
        e.stopImmediatePropagation();
      }
    };
    var clearStyleEl = function clearStyleEl() {
      clearTimeout(styleTimer.value);
      styleTimer.value = setTimeout(function () {
        var _styleEl$value, _styleEl$value$remove;
        (_styleEl$value = styleEl.value) === null || _styleEl$value === void 0 || (_styleEl$value = _styleEl$value.parentNode) === null || _styleEl$value === void 0 || (_styleEl$value$remove = _styleEl$value.removeChild) === null || _styleEl$value$remove === void 0 || _styleEl$value$remove.call(_styleEl$value, styleEl.value);
        styleEl.value = null;
      }, 150);
      Vue.nextTick(function () {
        var _drawerEle$value2, _drawerEle$value2$foc;
        (_drawerEle$value2 = drawerEle.value) === null || _drawerEle$value2 === void 0 || (_drawerEle$value2$foc = _drawerEle$value2.focus) === null || _drawerEle$value2$foc === void 0 || _drawerEle$value2$foc.call(_drawerEle$value2);
      });
    };
    var TRANSITION_CSS_MARGIN = "transition: margin 300ms cubic-bezier(0.7, 0.3, 0.1, 1) 0s;";
    var createStyleEl = function createStyleEl() {
      var hasScrollBar = window.innerWidth > document.documentElement.clientWidth;
      var scrollWidth = hasScrollBar ? getScrollbarWidth.getScrollbarWidth() : 0;
      styleEl.value = document.createElement("style");
      styleEl.value.dataset.id = "td_drawer_".concat(+new Date(), "_").concat(key += 1);
      styleEl.value.innerHTML = "\n        html body {\n          overflow-y: hidden;\n          ".concat(TRANSITION_CSS_MARGIN, "\n          ").concat(props2.mode === "push" ? "" : "width: calc(100% - ".concat(scrollWidth, "px);"), "\n        }\n      ");
    };
    var handlePushMode = function handlePushMode() {
      if (props2.mode !== "push") return;
      Vue.nextTick(function () {
        if (!parentNode.value) return;
        parentNode.value.style.cssText = TRANSITION_CSS_MARGIN;
      });
    };
    var updatePushMode = function updatePushMode() {
      if (!parentNode.value || props2.mode !== "push") return;
      var marginValueData = {
        left: {
          name: "margin-left",
          value: sizeValue.value
        },
        right: {
          name: "margin-left",
          value: "-".concat(sizeValue.value)
        },
        top: {
          name: "margin-top",
          value: sizeValue.value
        },
        bottom: {
          name: "margin-top",
          value: "-".concat(sizeValue.value)
        }
      }[props2.placement];
      if (isVisible.value) {
        parentNode.value.style.setProperty(marginValueData.name, marginValueData.value);
      } else {
        parentNode.value.style.removeProperty(marginValueData.name);
      }
    };
    var getDefaultFooter = function getDefaultFooter() {
      var confirmBtn = getConfirmBtn({
        confirmBtn: props2.confirmBtn,
        globalConfirm: globalConfig.value.confirm,
        className: "".concat(COMPONENT_NAME.value, "__confirm")
      });
      var cancelBtn = getCancelBtn({
        cancelBtn: props2.cancelBtn,
        globalCancel: globalConfig.value.cancel,
        className: "".concat(COMPONENT_NAME.value, "__cancel")
      });
      return Vue.createVNode("div", {
        "style": footerStyle.value
      }, [props2.placement === "right" ? confirmBtn : null, cancelBtn, props2.placement !== "right" ? confirmBtn : null]);
    };
    var _usePopupManager = index$3.usePopupManager("drawer", {
        visible: computedVisible
      }),
      isTopInteractivePopup = _usePopupManager.isTopInteractivePopup;
    Vue.watch(modeAndPlacement, handlePushMode, {
      immediate: true
    });
    var updateVisibleState = function updateVisibleState(value) {
      if (value) {
        isMounted.value = true;
      }
      if (props2.destroyOnClose) {
        if (value) {
          destroyOnCloseVisible.value = false;
          setTimeout(function () {
            return isVisible.value = true;
          });
        } else {
          isVisible.value = false;
          if (destroyOnCloseVisible.value) {
            destroyOnCloseVisible.value = false;
          }
          setTimeout(function () {
            return destroyOnCloseVisible.value = true;
          }, 300);
        }
        return;
      }
      if (destroyOnCloseVisible.value && value) {
        destroyOnCloseVisible.value = false;
        setTimeout(function () {
          return isVisible.value = true;
        });
        return;
      }
      setTimeout(function () {
        return isVisible.value = value;
      });
    };
    var addStyleElToHead = function addStyleElToHead() {
      if (!props2.showInAttachedElement && props2.preventScrollThrough && isVisible.value && (isMounted.value || !props2.lazy)) {
        if (!styleEl.value) {
          createStyleEl();
        }
        if (styleEl.value && !document.head.contains(styleEl.value)) {
          document.head.appendChild(styleEl.value);
        }
      }
    };
    Vue.watch(function () {
      return props2.visible;
    }, function (value) {
      if (dom.isServer) return;
      if (value) {
        var _props2$onBeforeOpen;
        addStyleElToHead();
        (_props2$onBeforeOpen = props2.onBeforeOpen) === null || _props2$onBeforeOpen === void 0 || _props2$onBeforeOpen.call(props2);
      } else {
        var _props2$onBeforeClose;
        clearStyleEl();
        (_props2$onBeforeClose = props2.onBeforeClose) === null || _props2$onBeforeClose === void 0 || _props2$onBeforeClose.call(props2);
      }
      updateVisibleState(value);
    }, {
      immediate: true
    });
    var handleCloseBtnClick = function handleCloseBtnClick(e) {
      var _props2$onCloseBtnCli;
      (_props2$onCloseBtnCli = props2.onCloseBtnClick) === null || _props2$onCloseBtnCli === void 0 || _props2$onCloseBtnCli.call(props2, {
        e: e
      });
      closeDrawer({
        trigger: "close-btn",
        e: e
      });
    };
    var handleWrapperClick = function handleWrapperClick(e) {
      var _props2$onOverlayClic, _props2$closeOnOverla;
      (_props2$onOverlayClic = props2.onOverlayClick) === null || _props2$onOverlayClic === void 0 || _props2$onOverlayClic.call(props2, {
        e: e
      });
      if ((_props2$closeOnOverla = props2.closeOnOverlayClick) !== null && _props2$closeOnOverla !== void 0 ? _props2$closeOnOverla : globalConfig.value.closeOnOverlayClick) {
        closeDrawer({
          trigger: "overlay",
          e: e
        });
      }
    };
    var closeDrawer = function closeDrawer(params) {
      var _props2$onClose;
      (_props2$onClose = props2.onClose) === null || _props2$onClose === void 0 || _props2$onClose.call(props2, params);
      context.emit("update:visible", false);
    };
    Vue.onUpdated(updatePushMode);
    Vue.onMounted(function () {
      addStyleElToHead();
      window.addEventListener("keydown", handleEscKeydown);
    });
    Vue.onBeforeUnmount(function () {
      clearStyleEl();
      window.removeEventListener("keydown", handleEscKeydown);
    });
    var shouldRender = Vue.computed(function () {
      if (!isMounted.value) {
        return !props2.lazy;
      } else {
        return isVisible.value || !destroyOnCloseVisible.value;
      }
    });
    return function () {
      if (!shouldRender.value) return;
      var body = renderContent("body", "default");
      var headerContent = renderTNodeJSX("header");
      var defaultFooter = getDefaultFooter();
      return Vue.createVNode(Vue.Teleport, {
        "disabled": !props2.attach || !teleportElement.value,
        "to": teleportElement.value
      }, {
        "default": function _default() {
          return [Vue.createVNode("div", Vue.mergeProps({
            "ref": drawerEle,
            "class": drawerClasses.value,
            "style": {
              zIndex: props2.zIndex
            },
            "onKeydown": handleEscKeydown,
            "tabindex": 0
          }, context.attrs), [props2.showOverlay && Vue.createVNode("div", {
            "class": "".concat(COMPONENT_NAME.value, "__mask"),
            "onClick": handleWrapperClick
          }, null), Vue.createVNode("div", {
            "class": wrapperClasses.value,
            "style": _objectSpread(_objectSpread({}, wrapperStyles.value), draggingStyles.value)
          }, [headerContent && Vue.createVNode("div", {
            "class": "".concat(COMPONENT_NAME.value, "__header")
          }, [headerContent]), props2.closeBtn && Vue.createVNode("div", {
            "class": "".concat(COMPONENT_NAME.value, "__close-btn"),
            "onClick": handleCloseBtnClick
          }, [renderTNodeJSX("closeBtn", Vue.createVNode(CloseIcon, null, null))]), Vue.createVNode("div", {
            "class": ["".concat(COMPONENT_NAME.value, "__body"), "narrow-scrollbar"]
          }, [body]), props2.footer && Vue.createVNode("div", {
            "class": "".concat(COMPONENT_NAME.value, "__footer")
          }, [renderTNodeJSX("footer", defaultFooter)]), props2.sizeDraggable && Vue.createVNode("div", {
            "style": draggableLineStyles.value,
            "onMousedown": enableDrag
          }, null)])])];
        }
      });
    };
  }
});

exports["default"] = _Drawer;
