// tdesign-vue-next/cjs/upload/components/multiple-flow-list.js
/**
 * tdesign v1.19.0
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _toConsumableArray = require('@babel/runtime/helpers/toConsumableArray');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var tdesignIconsVueNext = require('tdesign-icons-vue-next');
require('@babel/runtime/helpers/typeof');
require('../../_chunks/dep-5de13842.js');
var index = require('../../_chunks/dep-d522f199.js');
require('../../config-provider/hooks/useConfig.js');
var index$1 = require('../../_chunks/dep-bdb76501.js');
require('@babel/runtime/helpers/slicedToArray');
var utils = require('../../_chunks/dep-16969a9c.js');
require('../../_chunks/dep-31b1e4d7.js');
var index$2 = require('../../_chunks/dep-ce75db03.js');
var imageViewer_index = require('../../image-viewer/index.js');
var upload_constants_index = require('../constants/index.js');
var button_index = require('../../button/index.js');
var upload_hooks_useDrag = require('../hooks/useDrag.js');
var loading_index = require('../../loading/index.js');
var link_index = require('../../link/index.js');
var image_index = require('../../image/index.js');
var isFunction = require('../../_chunks/dep-f4e9436d.js');
require('../../_chunks/dep-a3769001.js');
require('../../_chunks/dep-ee9debba.js');
require('../../_chunks/dep-1ad2ac36.js');
require('../../_chunks/dep-04f0c113.js');
require('../../_chunks/dep-ac99965c.js');
require('../../_chunks/dep-c0f98bf1.js');
require('../../_chunks/dep-919c39e9.js');
require('../../_chunks/dep-e9164e20.js');
require('../../_chunks/dep-1f93a4fb.js');
require('../../_chunks/dep-8602102b.js');
require('../../_chunks/dep-36c3ff8a.js');
require('../../_chunks/dep-30b893ce.js');
require('../../_chunks/dep-29a6a0fe.js');
require('../../_chunks/dep-19fab953.js');
require('../../_chunks/dep-8408bcc7.js');
require('../../_chunks/dep-a9d730be.js');
require('dayjs');
require('../../_chunks/dep-eb5fcbdf.js');
require('../../_chunks/dep-83a70289.js');
require('../../_chunks/dep-5794541b.js');
require('../../_chunks/dep-b163b14a.js');
require('../../_chunks/dep-f3b0c370.js');
require('../../_chunks/dep-de6498e2.js');
require('../../_chunks/dep-20f60e28.js');
require('../../_chunks/dep-054cd750.js');
require('../../_chunks/dep-e708216c.js');
require('../../_chunks/dep-6d7cbeac.js');
require('../../_chunks/dep-49e8331f.js');
require('../../_chunks/dep-455c76a3.js');
require('../../_chunks/dep-ef0f6cb2.js');
require('../../_chunks/dep-4dca9fa4.js');
require('../../_chunks/dep-ab2c6a84.js');
require('../../_chunks/dep-fac69cec.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('../../image-viewer/image-viewer.js');
require('../../_chunks/dep-4212631b.js');
require('../../_chunks/dep-3ff65181.js');
require('../../_chunks/dep-182ccba7.js');
require('../../loading/icon/gradient.js');
require('../../_chunks/dep-d6320ff5.js');
require('../../_chunks/dep-c0d22f5a.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('../../_chunks/dep-0b181efe.js');
require('../../_chunks/dep-e2ef7eaa.js');
require('../../_chunks/dep-8e6a1720.js');
require('../../_chunks/dep-5fd85b1e.js');
require('../../loading/props.js');
require('../../_chunks/dep-721bc27c.js');
require('../../image-viewer/base/ImageItem.js');
require('@babel/runtime/helpers/asyncToGenerator');
require('@babel/runtime/regenerator');
require('../../_chunks/dep-5b0fdfa1.js');
require('../../image-viewer/hooks/index.js');
require('../../_chunks/dep-34227b99.js');
require('../../_chunks/dep-6dff652a.js');
require('../../_chunks/dep-ec8bee75.js');
require('../../_chunks/dep-3a21079e.js');
require('../../_chunks/dep-9c51e332.js');
require('../../image-viewer/base/ImageModalIcon.js');
require('../../image-viewer/base/ImageViewerModal.js');
require('../../dialog/index.js');
require('../../dialog/dialog.js');
require('../../dialog/props.js');
require('../../_chunks/dep-f807a184.js');
require('../../dialog/hooks/useSameTarget.js');
require('../../_chunks/dep-83f1b705.js');
require('../../dialog/utils/index.js');
require('../../dialog/dialog-card.js');
require('../../dialog/dialog-card-props.js');
require('../../dialog/hooks/useAction.js');
require('../../_chunks/dep-7325f3b1.js');
require('../../_chunks/dep-977d699f.js');
require('../../_chunks/dep-f55ada6c.js');
require('../../_chunks/dep-4511de03.js');
require('../../button/button.js');
require('../../button/props.js');
require('../../_chunks/dep-ec3949b4.js');
require('../../_chunks/dep-d40e34ba.js');
require('../../_chunks/dep-3c0b81a2.js');
require('../../_chunks/dep-f201a161.js');
require('../../loading/plugin.js');
require('../../_chunks/dep-2a185a5d.js');
require('../../_chunks/dep-a262c754.js');
require('../../_chunks/dep-7fa97010.js');
require('../../_chunks/dep-e75e2fe1.js');
require('../../_chunks/dep-761e41bd.js');
require('../../_chunks/dep-cb81cbfa.js');
require('../../_chunks/dep-4e47eae6.js');
require('../../_chunks/dep-566ac4ff.js');
require('../../_chunks/dep-6d02741a.js');
require('../../dialog/plugin.js');
require('../../image-viewer/props.js');
require('../../image-viewer/base/ImageViewerUtils.js');
require('../../tooltip/index.js');
require('../../tooltip/tooltip.js');
require('../../tooltip/props.js');
require('../../popup/props.js');
require('../../popup/index.js');
require('../../popup/popup.js');
require('@popperjs/core');
require('../../popup/container.js');
require('../../_chunks/dep-df0a6b30.js');
require('../../_chunks/dep-2a9e99e2.js');
require('../../_chunks/dep-ed830a31.js');
require('../../tooltip/utils/index.js');
require('../../image-viewer/constants/index.js');
require('../../image-viewer/utils/index.js');
require('../../image/image.js');
require('../../_chunks/dep-c3741c2c.js');
require('../../image/props.js');
require('../../space/index.js');
require('../../space/space.js');
require('../../space/props.js');
require('../../_chunks/dep-50d2328a.js');
require('../../link/link.js');
require('../../link/props.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _toConsumableArray__default = /*#__PURE__*/_interopDefaultLegacy(_toConsumableArray);
var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty__default["default"](e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _isSlot(s) {
  return typeof s === 'function' || Object.prototype.toString.call(s) === '[object Object]' && !Vue.isVNode(s);
}
var MultipleFlowList = Vue.defineComponent({
  name: "UploadMultipleFlowList",
  props: _objectSpread(_objectSpread({}, upload_constants_index.commonProps), {}, {
    showThumbnail: Boolean,
    uploadFiles: Function,
    cancelUpload: Function,
    dragEvents: Object,
    disabled: Boolean,
    isBatchUpload: Boolean,
    draggable: Boolean,
    showImageFileName: Boolean,
    uploadButton: Object,
    cancelUploadButton: Object,
    onPreview: Function
  }),
  setup: function setup(props, _ref) {
    var slots = _ref.slots;
    var _toRefs = Vue.toRefs(props),
      uploading = _toRefs.uploading,
      disabled = _toRefs.disabled,
      displayFiles = _toRefs.displayFiles,
      classPrefix = _toRefs.classPrefix,
      accept = _toRefs.accept;
    var uploadPrefix = Vue.computed(function () {
      return "".concat(classPrefix.value, "-upload");
    });
    var locale = Vue.computed(function () {
      return props.locale;
    });
    var renderTNodeJSX = index.useTNodeJSX();
    var _useGlobalIcon = index$1.useGlobalIcon({
        BrowseIcon: tdesignIconsVueNext.BrowseIcon,
        DeleteIcon: tdesignIconsVueNext.DeleteIcon,
        CheckCircleFilledIcon: tdesignIconsVueNext.CheckCircleFilledIcon,
        ErrorCircleFilledIcon: tdesignIconsVueNext.ErrorCircleFilledIcon,
        TimeFilledIcon: tdesignIconsVueNext.TimeFilledIcon,
        FileExcelIcon: tdesignIconsVueNext.FileExcelIcon,
        FilePdfIcon: tdesignIconsVueNext.FilePdfIcon,
        FileWordIcon: tdesignIconsVueNext.FileWordIcon,
        FilePowerpointIcon: tdesignIconsVueNext.FilePowerpointIcon,
        FileIcon: tdesignIconsVueNext.FileIcon,
        VideoIcon: tdesignIconsVueNext.VideoIcon
      }),
      BrowseIcon = _useGlobalIcon.BrowseIcon,
      DeleteIcon = _useGlobalIcon.DeleteIcon,
      CheckCircleFilledIcon = _useGlobalIcon.CheckCircleFilledIcon,
      ErrorCircleFilledIcon = _useGlobalIcon.ErrorCircleFilledIcon,
      TimeFilledIcon = _useGlobalIcon.TimeFilledIcon,
      FileExcelIcon = _useGlobalIcon.FileExcelIcon,
      FilePdfIcon = _useGlobalIcon.FilePdfIcon,
      FileWordIcon = _useGlobalIcon.FileWordIcon,
      FilePowerpointIcon = _useGlobalIcon.FilePowerpointIcon,
      FileIcon = _useGlobalIcon.FileIcon,
      VideoIcon = _useGlobalIcon.VideoIcon;
    var drag = upload_hooks_useDrag["default"](props.dragEvents, accept);
    var currentPreviewFile = Vue.ref([]);
    var previewIndex = Vue.ref(0);
    var uploadText = Vue.computed(function () {
      if (uploading.value) return "".concat(locale.value.progress.uploadingText);
      return locale.value.triggerUploadText.normal;
    });
    var innerDragEvents = Vue.computed(function () {
      var draggable = props.draggable === void 0 ? true : props.draggable;
      return draggable ? {
        onDrop: drag.handleDrop,
        onDragenter: drag.handleDragenter,
        onDragover: drag.handleDragover,
        onDragleave: drag.handleDragleave
      } : {};
    });
    var getStatusMap = function getStatusMap() {
      var iconMap = {
        success: Vue.createVNode(CheckCircleFilledIcon, null, null),
        fail: Vue.createVNode(ErrorCircleFilledIcon, null, null),
        progress: Vue.createVNode(loading_index.Loading, null, null),
        waiting: Vue.createVNode(TimeFilledIcon, null, null)
      };
      var progress = locale.value.progress;
      var textMap = {
        success: progress === null || progress === void 0 ? void 0 : progress.successText,
        fail: progress === null || progress === void 0 ? void 0 : progress.failText,
        progress: progress === null || progress === void 0 ? void 0 : progress.uploadingText,
        waiting: progress === null || progress === void 0 ? void 0 : progress.waitingText
      };
      return {
        iconMap: iconMap,
        textMap: textMap
      };
    };
    var renderEmpty = function renderEmpty() {
      return Vue.createVNode("div", {
        "class": "".concat(uploadPrefix.value, "__flow-empty")
      }, [drag.dragActive.value ? locale.value.dragger.dragDropText : locale.value.dragger.clickAndDragText]);
    };
    var renderImgItem = function renderImgItem(file, index) {
      var _file$response;
      var _getStatusMap = getStatusMap(),
        iconMap = _getStatusMap.iconMap,
        textMap = _getStatusMap.textMap;
      var fileName = props.abridgeName && file.name ? utils.abridgeName.apply(void 0, [file.name].concat(_toConsumableArray__default["default"](props.abridgeName))) : file.name;
      return Vue.createVNode("li", {
        "class": "".concat(uploadPrefix.value, "__card-item"),
        "key": file.name + index + file.percent + file.status || "0"
      }, [Vue.createVNode("div", {
        "class": ["".concat(uploadPrefix.value, "__card-content"), _defineProperty__default["default"]({}, "".concat(classPrefix.value, "-is-bordered"), file.status !== "waiting")]
      }, [file.status === "progress" && Vue.createVNode("div", {
        "class": "".concat(uploadPrefix.value, "__card-status-wrap ").concat(uploadPrefix.value, "__").concat(props.theme, "-progress")
      }, [iconMap[file.status], Vue.createVNode("p", null, [textMap[file.status], props.showUploadProgress && " ".concat(file.percent, "%")])]), file.status === "fail" && Vue.createVNode("div", {
        "class": "".concat(uploadPrefix.value, "__card-status-wrap ").concat(uploadPrefix.value, "__").concat(props.theme, "-fail")
      }, [iconMap[file.status], Vue.createVNode("p", null, [((_file$response = file.response) === null || _file$response === void 0 ? void 0 : _file$response.error) || textMap[file.status]])]), (["waiting", "success"].includes(file.status) || !file.status && file.url) && Vue.createVNode(image_index.Image, {
        "class": "".concat(uploadPrefix.value, "__card-image"),
        "src": file.url || file.raw,
        "error": "",
        "loading": "",
        "fit": "contain"
      }, null), Vue.createVNode("div", {
        "class": "".concat(uploadPrefix.value, "__card-mask")
      }, [(file.url || file.raw) && !["progress", "fail"].includes(file.status) && Vue.createVNode("span", {
        "class": "".concat(uploadPrefix.value, "__card-mask-item")
      }, [Vue.createVNode(BrowseIcon, {
        "onClick": function onClick(_ref3) {
          var _props$onPreview;
          var e = _ref3.e;
          previewIndex.value = index;
          currentPreviewFile.value = displayFiles.value;
          (_props$onPreview = props.onPreview) === null || _props$onPreview === void 0 || _props$onPreview.call(props, {
            file: file,
            index: index,
            e: e
          });
        }
      }, null), Vue.createVNode("span", {
        "class": "".concat(uploadPrefix.value, "__card-mask-item-divider")
      }, null)]), !disabled.value && Vue.createVNode("span", {
        "class": "".concat(uploadPrefix.value, "__card-mask-item ").concat(uploadPrefix.value, "__delete"),
        "onClick": function onClick(e) {
          return props.onRemove({
            e: e,
            index: index,
            file: file
          });
        }
      }, [Vue.createVNode(DeleteIcon, null, null)])])]), props.showImageFileName && Vue.createVNode("p", {
        "class": ["".concat(uploadPrefix.value, "__card-name"), "".concat(uploadPrefix.value, "__flow-status")]
      }, [["success", "waiting"].includes(file.status) && iconMap[file.status], fileName])]);
    };
    var renderStatus = function renderStatus(file) {
      var _file$response2, _file$response3;
      var _getStatusMap2 = getStatusMap(),
        iconMap = _getStatusMap2.iconMap,
        textMap = _getStatusMap2.textMap;
      return Vue.createVNode("div", {
        "class": "".concat(uploadPrefix.value, "__flow-status")
      }, [iconMap[file.status], Vue.createVNode("span", {
        "class": "".concat(uploadPrefix.value, "__").concat(props.theme, "-").concat(file.status)
      }, [(_file$response2 = file.response) !== null && _file$response2 !== void 0 && _file$response2.error ? ((_file$response3 = file.response) === null || _file$response3 === void 0 ? void 0 : _file$response3.error) || textMap[file.status] : textMap[file.status], props.showUploadProgress && file.status === "progress" ? " ".concat(file.percent || 0, "%") : ""])]);
    };
    var renderNormalActionCol = function renderNormalActionCol(file, index) {
      var _locale$value;
      return Vue.createVNode("td", null, [Vue.createVNode(button_index.Button, {
        "theme": "primary",
        "variant": "text",
        "content": (_locale$value = locale.value) === null || _locale$value === void 0 || (_locale$value = _locale$value.triggerUploadText) === null || _locale$value === void 0 ? void 0 : _locale$value["delete"],
        "class": "".concat(uploadPrefix.value, "__delete"),
        "onClick": function onClick(e) {
          return props.onRemove({
            e: e,
            index: index,
            file: file
          });
        }
      }, null)]);
    };
    var renderBatchActionCol = function renderBatchActionCol(index) {
      var _locale$value2;
      return index === 0 ? Vue.createVNode("td", {
        "rowspan": displayFiles.value.length,
        "class": "".concat(uploadPrefix.value, "__flow-table__batch-row")
      }, [Vue.createVNode(button_index.Button, {
        "theme": "primary",
        "variant": "text",
        "content": (_locale$value2 = locale.value) === null || _locale$value2 === void 0 || (_locale$value2 = _locale$value2.triggerUploadText) === null || _locale$value2 === void 0 ? void 0 : _locale$value2["delete"],
        "class": "".concat(uploadPrefix.value, "__delete"),
        "onClick": function onClick(e) {
          return props.onRemove({
            e: e,
            index: -1,
            file: void 0
          });
        }
      }, null)]) : null;
    };
    var getFileThumbnailIcon = function getFileThumbnailIcon(fileType) {
      if (utils.FILE_PDF_REGEXP.test(fileType)) {
        return Vue.createVNode(FilePdfIcon, null, null);
      }
      if (utils.FILE_EXCEL_REGEXP.test(fileType)) {
        return Vue.createVNode(FileExcelIcon, null, null);
      }
      if (utils.FILE_WORD_REGEXP.test(fileType)) {
        return Vue.createVNode(FileWordIcon, null, null);
      }
      if (utils.FILE_PPT_REGEXP.test(fileType)) {
        return Vue.createVNode(FilePowerpointIcon, null, null);
      }
      if (utils.VIDEO_REGEXP.test(fileType)) {
        return Vue.createVNode(VideoIcon, null, null);
      }
      return Vue.createVNode(FileIcon, null, null);
    };
    var renderFileThumbnail = function renderFileThumbnail(file) {
      if (!file || !file.raw && file.url) return null;
      var fileType = file.raw.type;
      var className = "".concat(uploadPrefix.value, "__file-thumbnail");
      if (utils.IMAGE_REGEXP.test(fileType)) {
        return Vue.createVNode("div", {
          "onClick": function onClick(e) {
            var _props$onPreview2;
            e.preventDefault();
            currentPreviewFile.value = [file];
            previewIndex.value = 0;
            (_props$onPreview2 = props.onPreview) === null || _props$onPreview2 === void 0 || _props$onPreview2.call(props, {
              file: file,
              index: 0,
              e: e
            });
          }
        }, [Vue.createVNode(image_index.Image, {
          "class": className,
          "src": file.url || file.raw,
          "fit": "scale-down",
          "error": "",
          "loading": ""
        }, null)]);
      }
      return Vue.createVNode("div", {
        "class": className
      }, [getFileThumbnailIcon(fileType)]);
    };
    var renderFileList = function renderFileList() {
      var _locale$value$file, _locale$value$file2, _locale$value$file3, _locale$value$file4;
      var customList = renderTNodeJSX("fileListDisplay", {
        params: {
          cancelUpload: props.cancelUpload,
          uploadFiles: props.uploadFiles,
          onPreview: props.onPreview,
          onRemove: props.onRemove,
          toUploadFiles: props.toUploadFiles,
          sizeOverLimitMessage: props.sizeOverLimitMessage,
          locale: props.locale,
          files: props.displayFiles,
          dragEvents: innerDragEvents.value
        }
      });
      if (customList || props.fileListDisplay) return customList;
      return Vue.createVNode("table", Vue.mergeProps({
        "class": "".concat(uploadPrefix.value, "__flow-table")
      }, innerDragEvents.value), [Vue.createVNode("thead", null, [Vue.createVNode("tr", null, [Vue.createVNode("th", null, [(_locale$value$file = locale.value.file) === null || _locale$value$file === void 0 ? void 0 : _locale$value$file.fileNameText]), Vue.createVNode("th", {
        "style": {
          minWidth: "120px"
        }
      }, [(_locale$value$file2 = locale.value.file) === null || _locale$value$file2 === void 0 ? void 0 : _locale$value$file2.fileSizeText]), Vue.createVNode("th", {
        "style": {
          minWidth: "120px"
        }
      }, [(_locale$value$file3 = locale.value.file) === null || _locale$value$file3 === void 0 ? void 0 : _locale$value$file3.fileStatusText]), disabled.value ? null : Vue.createVNode("th", null, [(_locale$value$file4 = locale.value.file) === null || _locale$value$file4 === void 0 ? void 0 : _locale$value$file4.fileOperationText])])]), Vue.createVNode("tbody", null, [!displayFiles.value.length && Vue.createVNode("tr", null, [Vue.createVNode("td", {
        "colspan": 4
      }, [renderEmpty()])]), displayFiles.value.map(function (file, index) {
        var _props$abridgeName;
        var showBatchUploadAction = props.isBatchUpload;
        var deleteNode = showBatchUploadAction && displayFiles.value.every(function (item) {
          return item.status === "success" || !item.status;
        }) ? renderBatchActionCol(index) : renderNormalActionCol(file, index);
        var fileName = (_props$abridgeName = props.abridgeName) !== null && _props$abridgeName !== void 0 && _props$abridgeName.length ? utils.abridgeName.apply(void 0, [file.name].concat(_toConsumableArray__default["default"](props.abridgeName))) : file.name;
        var thumbnailNode = props.showThumbnail ? Vue.createVNode("div", {
          "class": "".concat(uploadPrefix.value, "__file-info")
        }, [renderFileThumbnail(file), fileName]) : fileName;
        var fileNameNode = file.url ? Vue.createVNode(link_index.Link, {
          "href": file.url,
          "target": "_blank",
          "hover": "color",
          "disabled": false
        }, _isSlot(thumbnailNode) ? thumbnailNode : {
          "default": function _default() {
            return [thumbnailNode];
          }
        }) : thumbnailNode;
        return Vue.createVNode("tr", {
          "key": file.name + index + file.size
        }, [Vue.createVNode("td", {
          "class": "".concat(uploadPrefix.value, "__file-name"),
          "key": file.name + file.url
        }, [fileNameNode]), Vue.createVNode("td", null, [utils.returnFileSize(file.size)]), Vue.createVNode("td", null, [renderStatus(file)]), disabled.value ? null : deleteNode]);
      })])]);
    };
    var renderImageList = function renderImageList() {
      var customList = renderTNodeJSX("fileListDisplay", {
        params: {
          cancelUpload: props.cancelUpload,
          uploadFiles: props.uploadFiles,
          onRemove: props.onRemove,
          onPreview: props.onPreview,
          toUploadFiles: props.toUploadFiles,
          sizeOverLimitMessage: props.sizeOverLimitMessage,
          locale: props.locale,
          files: props.displayFiles,
          dragEvents: innerDragEvents.value
        }
      });
      if (customList || props.fileListDisplay) return customList;
      return Vue.createVNode("ul", {
        "class": "".concat(uploadPrefix.value, "__card clearfix")
      }, [props.displayFiles.map(function (file, index) {
        return renderImgItem(file, index);
      })]);
    };
    return function () {
      var _slots$default, _locale$value3, _locale$value4;
      var cardClassName = "".concat(uploadPrefix.value, "__flow-card-area");
      var cancelUploadDisabled = disabled.value || !uploading.value;
      var hasCancelUploadTNode = slots.uploadButton || isFunction.isFunction(props.uploadButton);
      var uploadButtonDisabled = Boolean(disabled.value || uploading.value || !displayFiles.value.length);
      var hasUploadButtonTNode = slots.cancelUploadButton || isFunction.isFunction(props.cancelUploadButton);
      var cancelButtonProps = index$2.useEventForward(props.cancelUploadButton, {
        onClick: function onClick(e) {
          var _props$cancelUpload;
          return (_props$cancelUpload = props.cancelUpload) === null || _props$cancelUpload === void 0 ? void 0 : _props$cancelUpload.call(props, {
            e: e
          });
        }
      });
      var uploadButtonProps = index$2.useEventForward(props.uploadButton, {
        onClick: function onClick() {
          var _props$uploadFiles;
          return (_props$uploadFiles = props.uploadFiles) === null || _props$uploadFiles === void 0 ? void 0 : _props$uploadFiles.call(props);
        }
      });
      return Vue.createVNode("div", {
        "class": "".concat(uploadPrefix.value, "__flow ").concat(uploadPrefix.value, "__flow-").concat(props.theme)
      }, [Vue.createVNode("div", {
        "class": "".concat(uploadPrefix.value, "__flow-op")
      }, [(_slots$default = slots["default"]) === null || _slots$default === void 0 ? void 0 : _slots$default.call(slots), props.placeholder && Vue.createVNode("small", {
        "class": "".concat(uploadPrefix.value, "__flow-placeholder ").concat(uploadPrefix.value, "__placeholder")
      }, [props.placeholder])]), props.theme === "image-flow" && Vue.createVNode("div", Vue.mergeProps({
        "class": cardClassName
      }, innerDragEvents.value), [displayFiles.value.length ? renderImageList() : renderEmpty()]), props.theme === "file-flow" && (displayFiles.value.length ? renderFileList() : Vue.createVNode("div", Vue.mergeProps({
        "class": cardClassName
      }, innerDragEvents.value), [renderEmpty()])), !props.autoUpload && (props.uploadButton !== null || props.cancelUploadButton !== null) && Vue.createVNode("div", {
        "class": "".concat(uploadPrefix.value, "__flow-bottom")
      }, [props.cancelUploadButton !== null && (hasCancelUploadTNode ? renderTNodeJSX("cancelUploadButton", {
        params: {
          disabled: cancelUploadDisabled,
          cancelUploadText: (_locale$value3 = locale.value) === null || _locale$value3 === void 0 ? void 0 : _locale$value3.cancelUploadText,
          cancelUpload: props.cancelUpload
        }
      }) : Vue.createVNode(button_index.Button, Vue.mergeProps({
        "theme": "default",
        "disabled": cancelUploadDisabled,
        "content": (_locale$value4 = locale.value) === null || _locale$value4 === void 0 ? void 0 : _locale$value4.cancelUploadText,
        "class": "".concat(uploadPrefix.value, "__cancel")
      }, cancelButtonProps.value), null)), props.uploadButton !== null && (hasUploadButtonTNode ? renderTNodeJSX("uploadButton", {
        params: {
          disabled: uploadButtonDisabled,
          uploading: uploading.value,
          uploadText: uploadText.value,
          uploadFiles: props.uploadFiles
        }
      }) : Vue.createVNode(button_index.Button, Vue.mergeProps({
        "disabled": uploadButtonDisabled,
        "theme": "primary",
        "loading": uploading.value,
        "class": "".concat(uploadPrefix.value, "__continue"),
        "content": uploadText.value
      }, uploadButtonProps.value), null))]), Vue.createVNode(imageViewer_index.ImageViewer, Vue.mergeProps({
        "images": currentPreviewFile.value.map(function (t) {
          return t.url || t.raw;
        }),
        "visible": !!currentPreviewFile.value.length,
        "onClose": function onClose() {
          currentPreviewFile.value = [];
        },
        "index": previewIndex.value,
        "onIndexChange": function onIndexChange(val) {
          return previewIndex.value = val;
        }
      }, props.imageViewerProps), null)]);
    };
  }
});

exports["default"] = MultipleFlowList;
