// tdesign-vue-next/cjs/transfer/components/transfer-list.js
/**
 * tdesign v1.19.0
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var _typeof = require('@babel/runtime/helpers/typeof');
var pagination_index = require('../../pagination/index.js');
var checkbox_index = require('../../checkbox/index.js');
var transfer_utils_index = require('../utils/index.js');
var transfer_components_transferSearch = require('./transfer-search.js');
require('@babel/runtime/helpers/toConsumableArray');
require('../../_chunks/dep-5de13842.js');
var index$1 = require('../../_chunks/dep-d522f199.js');
var index = require('../../_chunks/dep-4212631b.js');
require('@babel/runtime/helpers/slicedToArray');
require('../../_chunks/dep-31b1e4d7.js');
var transfer_hooks_useDragSort = require('../hooks/useDragSort.js');
var _arrayFilter = require('../../_chunks/dep-ab2c6a84.js');
var loading_directive = require('../../_chunks/dep-f201a161.js');
var isArrayLike = require('../../_chunks/dep-20f60e28.js');
var _baseIteratee = require('../../_chunks/dep-761e41bd.js');
var isArray = require('../../_chunks/dep-ac99965c.js');
var cloneDeep = require('../../_chunks/dep-455c76a3.js');
var isString = require('../../_chunks/dep-1ad2ac36.js');
var configProvider_hooks_useConfig = require('../../config-provider/hooks/useConfig.js');
require('../../pagination/pagination.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('tdesign-icons-vue-next');
require('../../_chunks/dep-3ff65181.js');
require('../../_chunks/dep-19fab953.js');
require('../../_chunks/dep-29a6a0fe.js');
require('../../_chunks/dep-1f93a4fb.js');
require('../../_chunks/dep-04f0c113.js');
require('../../_chunks/dep-8602102b.js');
require('../../_chunks/dep-bdb76501.js');
require('../../_chunks/dep-721bc27c.js');
require('../../input-number/index.js');
require('../../input-number/input-number.js');
require('../../button/index.js');
require('../../button/button.js');
require('../../loading/index.js');
require('../../_chunks/dep-182ccba7.js');
require('../../loading/icon/gradient.js');
require('../../_chunks/dep-d6320ff5.js');
require('../../_chunks/dep-c0d22f5a.js');
require('../../_chunks/dep-0b181efe.js');
require('../../_chunks/dep-e2ef7eaa.js');
require('../../_chunks/dep-8e6a1720.js');
require('../../_chunks/dep-ee9debba.js');
require('../../_chunks/dep-f4e9436d.js');
require('../../_chunks/dep-919c39e9.js');
require('../../_chunks/dep-5fd85b1e.js');
require('../../loading/props.js');
require('../../_chunks/dep-6d02741a.js');
require('../../loading/plugin.js');
require('../../_chunks/dep-8408bcc7.js');
require('../../_chunks/dep-a9d730be.js');
require('../../_chunks/dep-a3769001.js');
require('dayjs');
require('../../_chunks/dep-eb5fcbdf.js');
require('../../_chunks/dep-83a70289.js');
require('../../_chunks/dep-5794541b.js');
require('../../_chunks/dep-b163b14a.js');
require('../../_chunks/dep-f3b0c370.js');
require('../../_chunks/dep-de6498e2.js');
require('../../_chunks/dep-054cd750.js');
require('../../_chunks/dep-e708216c.js');
require('../../_chunks/dep-6d7cbeac.js');
require('../../_chunks/dep-49e8331f.js');
require('../../_chunks/dep-c0f98bf1.js');
require('../../_chunks/dep-e9164e20.js');
require('../../_chunks/dep-36c3ff8a.js');
require('../../_chunks/dep-30b893ce.js');
require('../../_chunks/dep-ef0f6cb2.js');
require('../../_chunks/dep-4dca9fa4.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('../../_chunks/dep-2a185a5d.js');
require('../../_chunks/dep-a262c754.js');
require('../../_chunks/dep-7fa97010.js');
require('../../_chunks/dep-e75e2fe1.js');
require('../../_chunks/dep-cb81cbfa.js');
require('../../_chunks/dep-977d699f.js');
require('../../_chunks/dep-4e47eae6.js');
require('../../_chunks/dep-566ac4ff.js');
require('../../button/props.js');
require('../../_chunks/dep-ec3949b4.js');
require('../../_chunks/dep-d40e34ba.js');
require('../../_chunks/dep-3c0b81a2.js');
require('../../input/index.js');
require('../../input/input.js');
require('../../input/props.js');
require('../../_chunks/dep-fd38990c.js');
require('../../input/hooks/useInput.js');
require('../../form/constants/index.js');
require('../../input/hooks/useLengthLimit.js');
require('../../_chunks/dep-fac69cec.js');
require('../../input/hooks/useInputEventHandler.js');
require('../../input/hooks/useInputWidth.js');
require('../../_chunks/dep-df0a6b30.js');
require('../../input/input-group.js');
require('../../input/input-group-props.js');
require('../../input-number/props.js');
require('../../_chunks/dep-ce75db03.js');
require('../../input-number/hooks/useInputNumber.js');
require('../../_chunks/dep-34227b99.js');
require('../../_chunks/dep-6dff652a.js');
require('../../select/index.js');
require('../../select/select.js');
require('@babel/runtime/helpers/asyncToGenerator');
require('@babel/runtime/regenerator');
require('../../common-components/fake-arrow.js');
require('../../select-input/index.js');
require('../../select-input/select-input.js');
require('../../popup/index.js');
require('../../popup/popup.js');
require('@popperjs/core');
require('../../popup/container.js');
require('../../popup/props.js');
require('../../_chunks/dep-2a9e99e2.js');
require('../../_chunks/dep-9c51e332.js');
require('../../_chunks/dep-ed830a31.js');
require('../../_chunks/dep-3a21079e.js');
require('../../select-input/props.js');
require('../../select-input/hooks/useMultiple.js');
require('../../tag-input/index.js');
require('../../tag-input/tag-input.js');
require('../../tag-input/props.js');
require('../../tag-input/hooks/useDragSorter.js');
require('../../tag-input/hooks/useHover.js');
require('../../tag-input/hooks/useTagScroll.js');
require('../../tag-input/hooks/useTagList.js');
require('../../tag/index.js');
require('../../tag/tag.js');
require('tinycolor2');
require('../../tag/props.js');
require('../../tag/check-tag.js');
require('../../tag/check-tag-props.js');
require('../../_chunks/dep-78278555.js');
require('../../tag/check-tag-group.js');
require('../../tag/check-tag-group-props.js');
require('../../select-input/hooks/useOverlayInnerStyle.js');
require('../../select-input/hooks/useSingle.js');
require('../../_chunks/dep-0e085d35.js');
require('../../_chunks/dep-7e2f9c1b.js');
require('../../_chunks/dep-4511de03.js');
require('../../select/components/select-panel.js');
require('../../select/option.js');
require('../../select/option-props.js');
require('../../select/utils/index.js');
require('../../select/constants/index.js');
require('../../checkbox/checkbox.js');
require('../../checkbox/props.js');
require('../../checkbox/constants/index.js');
require('../../checkbox/hooks/useCheckboxLazyLoad.js');
require('../../_chunks/dep-c3741c2c.js');
require('../../checkbox/hooks/useKeyboardEvent.js');
require('../../checkbox/group.js');
require('../../checkbox/checkbox-group-props.js');
require('../../_chunks/dep-50d2328a.js');
require('../../_chunks/dep-7f048e0f.js');
require('../../_chunks/dep-d5ee60d3.js');
require('../../select/option-group.js');
require('../../select/option-group-props.js');
require('../../select/props.js');
require('../../select/hooks/usePanelVirtualScroll.js');
require('../../_chunks/dep-7325f3b1.js');
require('../../_chunks/dep-f55ada6c.js');
require('../../select/hooks/useKeyboardControl.js');
require('../../select/hooks/useSelectOptions.js');
require('../../_chunks/dep-84b9a856.js');
require('../../input-adornment/index.js');
require('../../input-adornment/input-adornment.js');
require('../../input-adornment/props.js');
require('../../pagination/props.js');
require('../../pagination/hooks/useMoreAction.js');
require('../../pagination/hooks/usePaginationClasses.js');
require('../../pagination/pagination-mini.js');
require('../../pagination/pagination-mini-props.js');
require('../../tooltip/index.js');
require('../../tooltip/tooltip.js');
require('../../tooltip/props.js');
require('../../tooltip/utils/index.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);
var _typeof__default = /*#__PURE__*/_interopDefaultLegacy(_typeof);

/**
 * Creates a `baseEach` or `baseEachRight` function.
 *
 * @private
 * @param {Function} eachFunc The function to iterate over a collection.
 * @param {boolean} [fromRight] Specify iterating from right to left.
 * @returns {Function} Returns the new base function.
 */
function createBaseEach(eachFunc, fromRight) {
  return function (collection, iteratee) {
    if (collection == null) {
      return collection;
    }
    if (!isArrayLike.isArrayLike(collection)) {
      return eachFunc(collection, iteratee);
    }
    var length = collection.length,
      index = fromRight ? length : -1,
      iterable = Object(collection);
    while (fromRight ? index-- : ++index < length) {
      if (iteratee(iterable[index], index, iterable) === false) {
        break;
      }
    }
    return collection;
  };
}

/**
 * The base implementation of `_.forEach` without support for iteratee shorthands.
 *
 * @private
 * @param {Array|Object} collection The collection to iterate over.
 * @param {Function} iteratee The function invoked per iteration.
 * @returns {Array|Object} Returns `collection`.
 */
var baseEach = createBaseEach(loading_directive.baseForOwn);

/**
 * The base implementation of `_.filter` without support for iteratee shorthands.
 *
 * @private
 * @param {Array|Object} collection The collection to iterate over.
 * @param {Function} predicate The function invoked per iteration.
 * @returns {Array} Returns the new filtered array.
 */
function baseFilter(collection, predicate) {
  var result = [];
  baseEach(collection, function (value, index, collection) {
    if (predicate(value, index, collection)) {
      result.push(value);
    }
  });
  return result;
}

/**
 * Iterates over elements of `collection`, returning an array of all elements
 * `predicate` returns truthy for. The predicate is invoked with three
 * arguments: (value, index|key, collection).
 *
 * **Note:** Unlike `_.remove`, this method returns a new array.
 *
 * @static
 * @memberOf _
 * @since 0.1.0
 * @category Collection
 * @param {Array|Object} collection The collection to iterate over.
 * @param {Function} [predicate=_.identity] The function invoked per iteration.
 * @returns {Array} Returns the new filtered array.
 * @see _.reject
 * @example
 *
 * var users = [
 *   { 'user': 'barney', 'age': 36, 'active': true },
 *   { 'user': 'fred',   'age': 40, 'active': false }
 * ];
 *
 * _.filter(users, function(o) { return !o.active; });
 * // => objects for ['fred']
 *
 * // The `_.matches` iteratee shorthand.
 * _.filter(users, { 'age': 36, 'active': true });
 * // => objects for ['barney']
 *
 * // The `_.matchesProperty` iteratee shorthand.
 * _.filter(users, ['active', false]);
 * // => objects for ['fred']
 *
 * // The `_.property` iteratee shorthand.
 * _.filter(users, 'active');
 * // => objects for ['barney']
 *
 * // Combining several predicates using `_.overEvery` or `_.overSome`.
 * _.filter(users, _.overSome([{ 'age': 36 }, ['age', 40]]));
 * // => objects for ['fred', 'barney']
 */
function filter(collection, predicate) {
  var func = isArray.isArray(collection) ? _arrayFilter.arrayFilter : baseFilter;
  return func(collection, _baseIteratee.baseIteratee(predicate));
}

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty__default["default"](e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _isSlot(s) {
  return typeof s === 'function' || Object.prototype.toString.call(s) === '[object Object]' && !Vue.isVNode(s);
}
var props = {
  checkboxProps: {
    type: Object,
    "default": function _default() {
      return {};
    }
  },
  dataSource: {
    type: Array,
    "default": function _default() {
      return [];
    }
  },
  listType: {
    type: String,
    "default": "target"
  },
  title: {
    type: [String, Function]
  },
  checkedValue: {
    type: Array,
    "default": function _default() {
      return [];
    }
  },
  disabled: {
    type: Boolean,
    "default": false
  },
  search: {
    type: [Boolean, Object],
    "default": false
  },
  transferItem: Function,
  empty: {
    type: [Function, String]
  },
  pagination: [Boolean, Object],
  footer: [Function, String],
  checkAll: Boolean,
  isTreeMode: {
    type: Boolean,
    "default": false
  },
  onCheckedChange: Function,
  onPageChange: Function,
  onScroll: Function,
  onSearch: Function,
  onDataChange: Function,
  draggable: Boolean,
  currentValue: {
    type: Array
  }
};
var TransferList = Vue.defineComponent({
  name: "TTransferList",
  props: props,
  setup: function setup(props2) {
    var classPrefix = index.usePrefixClass();
    var _toRefs = Vue.toRefs(props2),
      currentValue = _toRefs.currentValue;
    var _useConfig = configProvider_hooks_useConfig.useConfig("transfer"),
      t = _useConfig.t,
      globalConfig = _useConfig.globalConfig;
    var filterValue = Vue.ref("");
    var defaultCurrent = Vue.ref(1);
    var defaultPageSize = Vue.ref(0);
    var currentPage = Vue.computed(function () {
      var pagination = props2.pagination;
      return (pagination === null || pagination === void 0 ? void 0 : pagination.current) || defaultCurrent.value || (pagination === null || pagination === void 0 ? void 0 : pagination.defaultCurrent);
    });
    var pageSize = Vue.computed(function () {
      var pagination = props2.pagination;
      return (pagination === null || pagination === void 0 ? void 0 : pagination.pageSize) || defaultPageSize.value || (pagination === null || pagination === void 0 ? void 0 : pagination.defaultPageSize);
    });
    var filteredData = Vue.computed(function () {
      var isTreeData = props2.dataSource.some(function (item) {
        return item.children && item.children.length;
      });
      if (!isTreeData) {
        return props2.dataSource.filter(function (item) {
          var label = item && item.label.toString();
          return label.toLowerCase().indexOf(filterValue.value.toLowerCase()) > -1;
        });
      } else {
        return _filteredTreeData(props2.dataSource, filterValue.value);
      }
    });
    var pageTotal = Vue.computed(function () {
      return filteredData.value && filteredData.value.length || 0;
    });
    var curPageData = Vue.computed(function () {
      var pageData = filteredData.value;
      if (!props2.pagination) return pageData;
      if (pageSize.value === 0) return pageData;
      var startIndex = (currentPage.value - 1) * pageSize.value;
      var endIndex = currentPage.value * pageSize.value;
      pageData = pageData.slice(startIndex, endIndex);
      return pageData;
    });
    var paginationProps = Vue.computed(function () {
      var defaultPaginationProps = {
        totalContent: false,
        pageSizeOptions: []
      };
      return _typeof__default["default"](props2.pagination) === "object" ? _objectSpread(_objectSpread(_objectSpread({}, defaultPaginationProps), props2.pagination), {}, {
        size: "small",
        theme: "simple",
        current: currentPage.value,
        total: pageTotal.value,
        pageSize: pageSize.value
      }) : {};
    });
    var _useDragSort = transfer_hooks_useDragSort["default"](currentValue, curPageData, props2.onDataChange),
      onDragStart = _useDragSort.onDragStart,
      onDragEnd = _useDragSort.onDragEnd,
      onDrop = _useDragSort.onDrop,
      onDragOver = _useDragSort.onDragOver,
      onDragLeave = _useDragSort.onDragLeave;
    var isAllChecked = Vue.computed(function () {
      var allValue = transfer_utils_index.getDataValues(props2.dataSource, [], {
        isTreeMode: props2.isTreeMode,
        include: false
      });
      return props2.checkedValue.length > 0 && (props2.isTreeMode ? allValue.every(function (item) {
        return props2.checkedValue.includes(item);
      }) : (props2.search ? filteredData.value : props2.dataSource).every(function (item) {
        return item.disabled || props2.checkedValue.includes(item.value);
      }));
    });
    var indeterminate = Vue.computed(function () {
      return !isAllChecked.value && props2.checkedValue.length > 0;
    });
    var totalCount = Vue.computed(function () {
      return transfer_utils_index.getLefCount(props2.dataSource);
    });
    Vue.watch(totalCount, function (val) {
      if (val <= (currentPage.value - 1) * pageSize.value) {
        var lastPage = Math.ceil(val / pageSize.value);
        defaultCurrent.value = lastPage;
      }
    });
    var _filteredTreeData = function filteredTreeData(list, keyword) {
      var res = filter(cloneDeep.cloneDeep(list), function (node) {
        if (node.label.toLowerCase().includes(keyword.toLowerCase())) {
          return true;
        }
        if (node.children && node.children.length > 0) {
          node.children = _filteredTreeData(node.children, keyword);
          if (node.children.length > 0) {
            return true;
          }
        }
        return false;
      });
      return res;
    };
    var handlePaginationChange = function handlePaginationChange(pageInfo) {
      var _props2$onPageChange;
      (_props2$onPageChange = props2.onPageChange) === null || _props2$onPageChange === void 0 || _props2$onPageChange.call(props2, pageInfo);
      defaultCurrent.value = pageInfo.current;
      defaultPageSize.value = pageInfo.pageSize;
    };
    var handleCheckedChange = function handleCheckedChange(val) {
      var _props2$onCheckedChan;
      (_props2$onCheckedChan = props2.onCheckedChange) === null || _props2$onCheckedChan === void 0 || _props2$onCheckedChan.call(props2, val);
    };
    var handleCheckedAllChange = function handleCheckedAllChange(checked) {
      if (checked) {
        var allValue = transfer_utils_index.getDataValues(props2.search ? filteredData.value : props2.dataSource, [], {
          isTreeMode: props2.isTreeMode,
          include: false
        });
        handleCheckedChange(allValue);
      } else {
        handleCheckedChange([]);
      }
    };
    var handleScroll = function handleScroll(e) {
      var _props2$onScroll;
      (_props2$onScroll = props2.onScroll) === null || _props2$onScroll === void 0 || _props2$onScroll.call(props2, e);
    };
    var handleSearch = function handleSearch(e) {
      var _props2$onSearch;
      var event = {
        query: e.value,
        type: props2.listType,
        e: e.e,
        trigger: e.trigger
      };
      filterValue.value = e.value;
      (_props2$onSearch = props2.onSearch) === null || _props2$onSearch === void 0 || _props2$onSearch.call(props2, event);
    };
    var renderTNodeJSX = index$1.useTNodeDefault();
    var renderTitle = function renderTitle() {
      var defaultNode = props2.title && isString.isString(props2.title) ? Vue.createVNode("template", null, [props2.title]) : null;
      var titleNode = renderTNodeJSX("title", {
        defaultNode: defaultNode,
        params: {
          type: props2.listType
        }
      });
      return Vue.createVNode("span", null, [titleNode]);
    };
    var renderContent = function renderContent() {
      var isDraggable = props2.draggable && props2.listType === transfer_utils_index.TARGET;
      var defaultNode = null;
      if (!isDraggable) {
        var _slot2;
        defaultNode = Vue.createVNode(checkbox_index.CheckboxGroup, {
          "value": props2.checkedValue,
          "onChange": handleCheckedChange
        }, _isSlot(_slot2 = curPageData.value.map(function (item, index) {
          var _slot;
          return Vue.createVNode(checkbox_index.Checkbox, Vue.mergeProps({
            "disabled": props2.disabled || item.disabled,
            "value": item.value,
            "needRipple": true,
            "class": ["".concat(classPrefix.value, "-transfer__list-item"), props2.checkedValue.includes(item.value) ? "".concat(classPrefix.value, "-is-checked") : ""],
            "key": item.key
          }, props2.checkboxProps), _isSlot(_slot = renderTNodeJSX("transferItem", {
            defaultNode: Vue.createVNode("span", null, [item.label]),
            params: {
              data: item.data,
              index: index,
              type: props2.listType
            }
          })) ? _slot : {
            "default": function _default() {
              return [_slot];
            }
          });
        })) ? _slot2 : {
          "default": function _default() {
            return [_slot2];
          }
        });
      } else {
        var _slot4;
        defaultNode = Vue.createVNode(checkbox_index.CheckboxGroup, {
          "value": props2.checkedValue,
          "onChange": handleCheckedChange
        }, _isSlot(_slot4 = curPageData.value.map(function (item, index) {
          var _slot3;
          return Vue.createVNode("div", {
            "draggable": isDraggable,
            "onDragend": onDragEnd,
            "onDragstart": onDragStart,
            "onDragover": onDragOver,
            "onDragleave": onDragLeave,
            "onDrop": onDrop,
            "data-index": index
          }, [Vue.createVNode(checkbox_index.Checkbox, Vue.mergeProps({
            "disabled": props2.disabled || item.disabled,
            "value": item.value,
            "needRipple": true,
            "class": ["".concat(classPrefix.value, "-transfer__list-item"), props2.checkedValue.includes(item.value) ? "".concat(classPrefix.value, "-is-checked") : ""],
            "key": item.key
          }, props2.checkboxProps), _isSlot(_slot3 = renderTNodeJSX("transferItem", {
            defaultNode: Vue.createVNode("span", null, [item.label]),
            params: {
              data: item.data,
              index: index,
              type: props2.listType
            }
          })) ? _slot3 : {
            "default": function _default() {
              return [_slot3];
            }
          })]);
        })) ? _slot4 : {
          "default": function _default() {
            return [_slot4];
          }
        });
      }
      return Vue.createVNode("div", {
        "class": ["".concat(classPrefix.value, "-transfer__list-content"), "narrow-scrollbar"],
        "onScroll": handleScroll
      }, [renderTNodeJSX("tree", {
        defaultNode: defaultNode,
        params: {
          data: curPageData.value,
          value: props2.checkedValue,
          onChange: handleCheckedChange
        }
      })]);
    };
    var renderEmpty = function renderEmpty() {
      var empty = props2.empty || t(globalConfig.value.empty);
      var defaultNode = isString.isString(empty) ? Vue.createVNode("span", null, [empty]) : null;
      return Vue.createVNode("div", {
        "class": "".concat(classPrefix.value, "-transfer__empty")
      }, [renderTNodeJSX("empty", {
        defaultNode: defaultNode,
        params: {
          type: props2.listType
        }
      })]);
    };
    var renderFooter = function renderFooter() {
      var defaultNode = isString.isString(props2.footer) ? Vue.createVNode("div", {
        "class": "".concat(classPrefix.value, "-transfer__footer")
      }, [props2.footer]) : null;
      return renderTNodeJSX("footer", {
        defaultNode: defaultNode,
        params: {
          type: props2.listType
        }
      });
    };
    return function () {
      return Vue.createVNode("div", {
        "class": ["".concat(classPrefix.value, "-transfer__list"), "".concat(classPrefix.value, "-transfer__list-").concat(props2.listType)]
      }, [Vue.createVNode("div", {
        "class": "".concat(classPrefix.value, "-transfer__list-header")
      }, [Vue.createVNode("div", null, [props2.checkAll && Vue.createVNode(checkbox_index.Checkbox, {
        "disabled": props2.disabled || !props2.dataSource.length,
        "checked": isAllChecked.value,
        "indeterminate": indeterminate.value,
        "onChange": handleCheckedAllChange
      }, null), Vue.createVNode("span", null, [t(globalConfig.value.title, {
        checked: props2.checkedValue.length,
        total: totalCount.value
      })])]), renderTitle()]), Vue.createVNode("div", {
        "class": ["".concat(classPrefix.value, "-transfer__list-body"), props2.search ? "".concat(classPrefix.value, "-transfer__list--with-search") : ""]
      }, [props2.search && Vue.createVNode(transfer_components_transferSearch["default"], {
        "value": filterValue.value,
        "placeholder": t(globalConfig.value.placeholder),
        "onChange": handleSearch,
        "search": props2.search
      }, null), curPageData.value.length > 0 ? renderContent() : renderEmpty()]), props2.pagination && pageSize.value > 0 && pageTotal.value > 0 && Vue.createVNode("div", {
        "class": "".concat(classPrefix.value, "-transfer__list-pagination")
      }, [Vue.createVNode(pagination_index.Pagination, Vue.mergeProps(paginationProps.value, {
        "onChange": handlePaginationChange
      }), null)]), renderFooter()]);
    };
  }
});

exports["default"] = TransferList;
